// json-editor-vue/dist/json-editor-vue.mjs
import { isVue3 as Ct, defineComponent as cn, getCurrentInstance as un, ref as qt, onUnmounted as sn, onMounted as fn, watch as tt, unref as ln, h as pn } from "vue-demi";
import { JSONEditor as vn } from "vanilla-jsoneditor";
var dn = typeof global == "object" && global && global.Object === Object && global;
const Sr = dn;
var bn = typeof self == "object" && self && self.Object === Object && self, yn = Sr || bn || Function("return this")();
const T = yn;
var hn = T.Symbol;
const S = hn;
var xr = Object.prototype, jn = xr.hasOwnProperty, gn = xr.toString, V = S ? S.toStringTag : void 0;
function mn(t) {
  var r = jn.call(t, V), n = t[V];
  try {
    t[V] = void 0;
    var e = !0;
  } catch {
  }
  var o = gn.call(t);
  return e && (r ? t[V] = n : delete t[V]), o;
}
var _n = Object.prototype, On = _n.toString;
function wn(t) {
  return On.call(t);
}
var An = "[object Null]", Sn = "[object Undefined]", Gt = S ? S.toStringTag : void 0;
function $(t) {
  return t == null ? t === void 0 ? Sn : An : Gt && Gt in Object(t) ? mn(t) : wn(t);
}
function x(t) {
  return t != null && typeof t == "object";
}
var xn = "[object Symbol]";
function It(t) {
  return typeof t == "symbol" || x(t) && $(t) == xn;
}
function Tn(t, r) {
  for (var n = -1, e = t == null ? 0 : t.length, o = Array(e); ++n < e; )
    o[n] = r(t[n], n, t);
  return o;
}
var En = Array.isArray;
const O = En;
var Cn = 1 / 0, Kt = S ? S.prototype : void 0, Jt = Kt ? Kt.toString : void 0;
function Tr(t) {
  if (typeof t == "string")
    return t;
  if (O(t))
    return Tn(t, Tr) + "";
  if (It(t))
    return Jt ? Jt.call(t) : "";
  var r = t + "";
  return r == "0" && 1 / t == -Cn ? "-0" : r;
}
function E(t) {
  var r = typeof t;
  return t != null && (r == "object" || r == "function");
}
function Pt(t) {
  return t;
}
var In = "[object AsyncFunction]", Pn = "[object Function]", kn = "[object GeneratorFunction]", Fn = "[object Proxy]";
function kt(t) {
  if (!E(t))
    return !1;
  var r = $(t);
  return r == Pn || r == kn || r == In || r == Fn;
}
var zn = T["__core-js_shared__"];
const ht = zn;
var Ht = function() {
  var t = /[^.]+$/.exec(ht && ht.keys && ht.keys.IE_PROTO || "");
  return t ? "Symbol(src)_1." + t : "";
}();
function $n(t) {
  return !!Ht && Ht in t;
}
var Ln = Function.prototype, Bn = Ln.toString;
function L(t) {
  if (t != null) {
    try {
      return Bn.call(t);
    } catch {
    }
    try {
      return t + "";
    } catch {
    }
  }
  return "";
}
var Un = /[\\^$.*+?()[\]{}|]/g, Mn = /^\[object .+?Constructor\]$/, Dn = Function.prototype, Nn = Object.prototype, Rn = Dn.toString, Wn = Nn.hasOwnProperty, Vn = RegExp(
  "^" + Rn.call(Wn).replace(Un, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function qn(t) {
  if (!E(t) || $n(t))
    return !1;
  var r = kt(t) ? Vn : Mn;
  return r.test(L(t));
}
function Gn(t, r) {
  return t == null ? void 0 : t[r];
}
function B(t, r) {
  var n = Gn(t, r);
  return qn(n) ? n : void 0;
}
var Kn = B(T, "WeakMap");
const _t = Kn;
var Xt = Object.create, Jn = /* @__PURE__ */ function() {
  function t() {
  }
  return function(r) {
    if (!E(r))
      return {};
    if (Xt)
      return Xt(r);
    t.prototype = r;
    var n = new t();
    return t.prototype = void 0, n;
  };
}();
const Hn = Jn;
function Xn(t, r, n) {
  switch (n.length) {
    case 0:
      return t.call(r);
    case 1:
      return t.call(r, n[0]);
    case 2:
      return t.call(r, n[0], n[1]);
    case 3:
      return t.call(r, n[0], n[1], n[2]);
  }
  return t.apply(r, n);
}
function Er(t, r) {
  var n = -1, e = t.length;
  for (r || (r = Array(e)); ++n < e; )
    r[n] = t[n];
  return r;
}
var Qn = 800, Zn = 16, Yn = Date.now;
function te(t) {
  var r = 0, n = 0;
  return function() {
    var e = Yn(), o = Zn - (e - n);
    if (n = e, o > 0) {
      if (++r >= Qn)
        return arguments[0];
    } else
      r = 0;
    return t.apply(void 0, arguments);
  };
}
function re(t) {
  return function() {
    return t;
  };
}
var ne = function() {
  try {
    var t = B(Object, "defineProperty");
    return t({}, "", {}), t;
  } catch {
  }
}();
const ct = ne;
var ee = ct ? function(t, r) {
  return ct(t, "toString", {
    configurable: !0,
    enumerable: !1,
    value: re(r),
    writable: !0
  });
} : Pt;
const oe = ee;
var ie = te(oe);
const ae = ie;
function ce(t, r) {
  for (var n = -1, e = t == null ? 0 : t.length; ++n < e && r(t[n], n, t) !== !1; )
    ;
  return t;
}
var ue = 9007199254740991, se = /^(?:0|[1-9]\d*)$/;
function Ft(t, r) {
  var n = typeof t;
  return r = r ?? ue, !!r && (n == "number" || n != "symbol" && se.test(t)) && t > -1 && t % 1 == 0 && t < r;
}
function pt(t, r, n) {
  r == "__proto__" && ct ? ct(t, r, {
    configurable: !0,
    enumerable: !0,
    value: n,
    writable: !0
  }) : t[r] = n;
}
function Z(t, r) {
  return t === r || t !== t && r !== r;
}
var fe = Object.prototype, le = fe.hasOwnProperty;
function Cr(t, r, n) {
  var e = t[r];
  (!(le.call(t, r) && Z(e, n)) || n === void 0 && !(r in t)) && pt(t, r, n);
}
function N(t, r, n, e) {
  var o = !n;
  n || (n = {});
  for (var i = -1, a = r.length; ++i < a; ) {
    var c = r[i], u = e ? e(n[c], t[c], c, n, t) : void 0;
    u === void 0 && (u = t[c]), o ? pt(n, c, u) : Cr(n, c, u);
  }
  return n;
}
var Qt = Math.max;
function pe(t, r, n) {
  return r = Qt(r === void 0 ? t.length - 1 : r, 0), function() {
    for (var e = arguments, o = -1, i = Qt(e.length - r, 0), a = Array(i); ++o < i; )
      a[o] = e[r + o];
    o = -1;
    for (var c = Array(r + 1); ++o < r; )
      c[o] = e[o];
    return c[r] = n(a), Xn(t, this, c);
  };
}
function ve(t, r) {
  return ae(pe(t, r, Pt), t + "");
}
var de = 9007199254740991;
function zt(t) {
  return typeof t == "number" && t > -1 && t % 1 == 0 && t <= de;
}
function vt(t) {
  return t != null && zt(t.length) && !kt(t);
}
function be(t, r, n) {
  if (!E(n))
    return !1;
  var e = typeof r;
  return (e == "number" ? vt(n) && Ft(r, n.length) : e == "string" && r in n) ? Z(n[r], t) : !1;
}
function Ir(t) {
  return ve(function(r, n) {
    var e = -1, o = n.length, i = o > 1 ? n[o - 1] : void 0, a = o > 2 ? n[2] : void 0;
    for (i = t.length > 3 && typeof i == "function" ? (o--, i) : void 0, a && be(n[0], n[1], a) && (i = o < 3 ? void 0 : i, o = 1), r = Object(r); ++e < o; ) {
      var c = n[e];
      c && t(r, c, e, i);
    }
    return r;
  });
}
var ye = Object.prototype;
function $t(t) {
  var r = t && t.constructor, n = typeof r == "function" && r.prototype || ye;
  return t === n;
}
function he(t, r) {
  for (var n = -1, e = Array(t); ++n < t; )
    e[n] = r(n);
  return e;
}
var je = "[object Arguments]";
function Zt(t) {
  return x(t) && $(t) == je;
}
var Pr = Object.prototype, ge = Pr.hasOwnProperty, me = Pr.propertyIsEnumerable, _e = Zt(/* @__PURE__ */ function() {
  return arguments;
}()) ? Zt : function(t) {
  return x(t) && ge.call(t, "callee") && !me.call(t, "callee");
};
const ut = _e;
function Oe() {
  return !1;
}
var kr = typeof exports == "object" && exports && !exports.nodeType && exports, Yt = kr && typeof module == "object" && module && !module.nodeType && module, we = Yt && Yt.exports === kr, tr = we ? T.Buffer : void 0, Ae = tr ? tr.isBuffer : void 0, Se = Ae || Oe;
const J = Se;
var xe = "[object Arguments]", Te = "[object Array]", Ee = "[object Boolean]", Ce = "[object Date]", Ie = "[object Error]", Pe = "[object Function]", ke = "[object Map]", Fe = "[object Number]", ze = "[object Object]", $e = "[object RegExp]", Le = "[object Set]", Be = "[object String]", Ue = "[object WeakMap]", Me = "[object ArrayBuffer]", De = "[object DataView]", Ne = "[object Float32Array]", Re = "[object Float64Array]", We = "[object Int8Array]", Ve = "[object Int16Array]", qe = "[object Int32Array]", Ge = "[object Uint8Array]", Ke = "[object Uint8ClampedArray]", Je = "[object Uint16Array]", He = "[object Uint32Array]", j = {};
j[Ne] = j[Re] = j[We] = j[Ve] = j[qe] = j[Ge] = j[Ke] = j[Je] = j[He] = !0;
j[xe] = j[Te] = j[Me] = j[Ee] = j[De] = j[Ce] = j[Ie] = j[Pe] = j[ke] = j[Fe] = j[ze] = j[$e] = j[Le] = j[Be] = j[Ue] = !1;
function Xe(t) {
  return x(t) && zt(t.length) && !!j[$(t)];
}
function Lt(t) {
  return function(r) {
    return t(r);
  };
}
var Fr = typeof exports == "object" && exports && !exports.nodeType && exports, G = Fr && typeof module == "object" && module && !module.nodeType && module, Qe = G && G.exports === Fr, jt = Qe && Sr.process, Ze = function() {
  try {
    var t = G && G.require && G.require("util").types;
    return t || jt && jt.binding && jt.binding("util");
  } catch {
  }
}();
const D = Ze;
var rr = D && D.isTypedArray, Ye = rr ? Lt(rr) : Xe;
const Bt = Ye;
var to = Object.prototype, ro = to.hasOwnProperty;
function zr(t, r) {
  var n = O(t), e = !n && ut(t), o = !n && !e && J(t), i = !n && !e && !o && Bt(t), a = n || e || o || i, c = a ? he(t.length, String) : [], u = c.length;
  for (var f in t)
    (r || ro.call(t, f)) && !(a && // Safari 9 has enumerable `arguments.length` in strict mode.
    (f == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    o && (f == "offset" || f == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    i && (f == "buffer" || f == "byteLength" || f == "byteOffset") || // Skip index properties.
    Ft(f, u))) && c.push(f);
  return c;
}
function $r(t, r) {
  return function(n) {
    return t(r(n));
  };
}
var no = $r(Object.keys, Object);
const eo = no;
var oo = Object.prototype, io = oo.hasOwnProperty;
function ao(t) {
  if (!$t(t))
    return eo(t);
  var r = [];
  for (var n in Object(t))
    io.call(t, n) && n != "constructor" && r.push(n);
  return r;
}
function Y(t) {
  return vt(t) ? zr(t) : ao(t);
}
function co(t) {
  var r = [];
  if (t != null)
    for (var n in Object(t))
      r.push(n);
  return r;
}
var uo = Object.prototype, so = uo.hasOwnProperty;
function fo(t) {
  if (!E(t))
    return co(t);
  var r = $t(t), n = [];
  for (var e in t)
    e == "constructor" && (r || !so.call(t, e)) || n.push(e);
  return n;
}
function R(t) {
  return vt(t) ? zr(t, !0) : fo(t);
}
var lo = Ir(function(t, r, n, e) {
  N(r, R(r), t, e);
});
const po = lo;
var vo = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, bo = /^\w*$/;
function Ut(t, r) {
  if (O(t))
    return !1;
  var n = typeof t;
  return n == "number" || n == "symbol" || n == "boolean" || t == null || It(t) ? !0 : bo.test(t) || !vo.test(t) || r != null && t in Object(r);
}
var yo = B(Object, "create");
const H = yo;
function ho() {
  this.__data__ = H ? H(null) : {}, this.size = 0;
}
function jo(t) {
  var r = this.has(t) && delete this.__data__[t];
  return this.size -= r ? 1 : 0, r;
}
var go = "__lodash_hash_undefined__", mo = Object.prototype, _o = mo.hasOwnProperty;
function Oo(t) {
  var r = this.__data__;
  if (H) {
    var n = r[t];
    return n === go ? void 0 : n;
  }
  return _o.call(r, t) ? r[t] : void 0;
}
var wo = Object.prototype, Ao = wo.hasOwnProperty;
function So(t) {
  var r = this.__data__;
  return H ? r[t] !== void 0 : Ao.call(r, t);
}
var xo = "__lodash_hash_undefined__";
function To(t, r) {
  var n = this.__data__;
  return this.size += this.has(t) ? 0 : 1, n[t] = H && r === void 0 ? xo : r, this;
}
function z(t) {
  var r = -1, n = t == null ? 0 : t.length;
  for (this.clear(); ++r < n; ) {
    var e = t[r];
    this.set(e[0], e[1]);
  }
}
z.prototype.clear = ho;
z.prototype.delete = jo;
z.prototype.get = Oo;
z.prototype.has = So;
z.prototype.set = To;
function Eo() {
  this.__data__ = [], this.size = 0;
}
function dt(t, r) {
  for (var n = t.length; n--; )
    if (Z(t[n][0], r))
      return n;
  return -1;
}
var Co = Array.prototype, Io = Co.splice;
function Po(t) {
  var r = this.__data__, n = dt(r, t);
  if (n < 0)
    return !1;
  var e = r.length - 1;
  return n == e ? r.pop() : Io.call(r, n, 1), --this.size, !0;
}
function ko(t) {
  var r = this.__data__, n = dt(r, t);
  return n < 0 ? void 0 : r[n][1];
}
function Fo(t) {
  return dt(this.__data__, t) > -1;
}
function zo(t, r) {
  var n = this.__data__, e = dt(n, t);
  return e < 0 ? (++this.size, n.push([t, r])) : n[e][1] = r, this;
}
function C(t) {
  var r = -1, n = t == null ? 0 : t.length;
  for (this.clear(); ++r < n; ) {
    var e = t[r];
    this.set(e[0], e[1]);
  }
}
C.prototype.clear = Eo;
C.prototype.delete = Po;
C.prototype.get = ko;
C.prototype.has = Fo;
C.prototype.set = zo;
var $o = B(T, "Map");
const X = $o;
function Lo() {
  this.size = 0, this.__data__ = {
    hash: new z(),
    map: new (X || C)(),
    string: new z()
  };
}
function Bo(t) {
  var r = typeof t;
  return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? t !== "__proto__" : t === null;
}
function bt(t, r) {
  var n = t.__data__;
  return Bo(r) ? n[typeof r == "string" ? "string" : "hash"] : n.map;
}
function Uo(t) {
  var r = bt(this, t).delete(t);
  return this.size -= r ? 1 : 0, r;
}
function Mo(t) {
  return bt(this, t).get(t);
}
function Do(t) {
  return bt(this, t).has(t);
}
function No(t, r) {
  var n = bt(this, t), e = n.size;
  return n.set(t, r), this.size += n.size == e ? 0 : 1, this;
}
function I(t) {
  var r = -1, n = t == null ? 0 : t.length;
  for (this.clear(); ++r < n; ) {
    var e = t[r];
    this.set(e[0], e[1]);
  }
}
I.prototype.clear = Lo;
I.prototype.delete = Uo;
I.prototype.get = Mo;
I.prototype.has = Do;
I.prototype.set = No;
var Ro = "Expected a function";
function Mt(t, r) {
  if (typeof t != "function" || r != null && typeof r != "function")
    throw new TypeError(Ro);
  var n = function() {
    var e = arguments, o = r ? r.apply(this, e) : e[0], i = n.cache;
    if (i.has(o))
      return i.get(o);
    var a = t.apply(this, e);
    return n.cache = i.set(o, a) || i, a;
  };
  return n.cache = new (Mt.Cache || I)(), n;
}
Mt.Cache = I;
var Wo = 500;
function Vo(t) {
  var r = Mt(t, function(e) {
    return n.size === Wo && n.clear(), e;
  }), n = r.cache;
  return r;
}
var qo = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, Go = /\\(\\)?/g, Ko = Vo(function(t) {
  var r = [];
  return t.charCodeAt(0) === 46 && r.push(""), t.replace(qo, function(n, e, o, i) {
    r.push(o ? i.replace(Go, "$1") : e || n);
  }), r;
});
const Jo = Ko;
function Ho(t) {
  return t == null ? "" : Tr(t);
}
function Lr(t, r) {
  return O(t) ? t : Ut(t, r) ? [t] : Jo(Ho(t));
}
var Xo = 1 / 0;
function yt(t) {
  if (typeof t == "string" || It(t))
    return t;
  var r = t + "";
  return r == "0" && 1 / t == -Xo ? "-0" : r;
}
function Br(t, r) {
  r = Lr(r, t);
  for (var n = 0, e = r.length; t != null && n < e; )
    t = t[yt(r[n++])];
  return n && n == e ? t : void 0;
}
function Qo(t, r, n) {
  var e = t == null ? void 0 : Br(t, r);
  return e === void 0 ? n : e;
}
function Ur(t, r) {
  for (var n = -1, e = r.length, o = t.length; ++n < e; )
    t[o + n] = r[n];
  return t;
}
var Zo = $r(Object.getPrototypeOf, Object);
const Dt = Zo;
var Yo = "[object Object]", ti = Function.prototype, ri = Object.prototype, Mr = ti.toString, ni = ri.hasOwnProperty, ei = Mr.call(Object);
function Dr(t) {
  if (!x(t) || $(t) != Yo)
    return !1;
  var r = Dt(t);
  if (r === null)
    return !0;
  var n = ni.call(r, "constructor") && r.constructor;
  return typeof n == "function" && n instanceof n && Mr.call(n) == ei;
}
function oi() {
  this.__data__ = new C(), this.size = 0;
}
function ii(t) {
  var r = this.__data__, n = r.delete(t);
  return this.size = r.size, n;
}
function ai(t) {
  return this.__data__.get(t);
}
function ci(t) {
  return this.__data__.has(t);
}
var ui = 200;
function si(t, r) {
  var n = this.__data__;
  if (n instanceof C) {
    var e = n.__data__;
    if (!X || e.length < ui - 1)
      return e.push([t, r]), this.size = ++n.size, this;
    n = this.__data__ = new I(e);
  }
  return n.set(t, r), this.size = n.size, this;
}
function w(t) {
  var r = this.__data__ = new C(t);
  this.size = r.size;
}
w.prototype.clear = oi;
w.prototype.delete = ii;
w.prototype.get = ai;
w.prototype.has = ci;
w.prototype.set = si;
function fi(t, r) {
  return t && N(r, Y(r), t);
}
function li(t, r) {
  return t && N(r, R(r), t);
}
var Nr = typeof exports == "object" && exports && !exports.nodeType && exports, nr = Nr && typeof module == "object" && module && !module.nodeType && module, pi = nr && nr.exports === Nr, er = pi ? T.Buffer : void 0, or = er ? er.allocUnsafe : void 0;
function Rr(t, r) {
  if (r)
    return t.slice();
  var n = t.length, e = or ? or(n) : new t.constructor(n);
  return t.copy(e), e;
}
function vi(t, r) {
  for (var n = -1, e = t == null ? 0 : t.length, o = 0, i = []; ++n < e; ) {
    var a = t[n];
    r(a, n, t) && (i[o++] = a);
  }
  return i;
}
function Wr() {
  return [];
}
var di = Object.prototype, bi = di.propertyIsEnumerable, ir = Object.getOwnPropertySymbols, yi = ir ? function(t) {
  return t == null ? [] : (t = Object(t), vi(ir(t), function(r) {
    return bi.call(t, r);
  }));
} : Wr;
const Nt = yi;
function hi(t, r) {
  return N(t, Nt(t), r);
}
var ji = Object.getOwnPropertySymbols, gi = ji ? function(t) {
  for (var r = []; t; )
    Ur(r, Nt(t)), t = Dt(t);
  return r;
} : Wr;
const Vr = gi;
function mi(t, r) {
  return N(t, Vr(t), r);
}
function qr(t, r, n) {
  var e = r(t);
  return O(t) ? e : Ur(e, n(t));
}
function Ot(t) {
  return qr(t, Y, Nt);
}
function _i(t) {
  return qr(t, R, Vr);
}
var Oi = B(T, "DataView");
const wt = Oi;
var wi = B(T, "Promise");
const At = wi;
var Ai = B(T, "Set");
const St = Ai;
var ar = "[object Map]", Si = "[object Object]", cr = "[object Promise]", ur = "[object Set]", sr = "[object WeakMap]", fr = "[object DataView]", xi = L(wt), Ti = L(X), Ei = L(At), Ci = L(St), Ii = L(_t), F = $;
(wt && F(new wt(new ArrayBuffer(1))) != fr || X && F(new X()) != ar || At && F(At.resolve()) != cr || St && F(new St()) != ur || _t && F(new _t()) != sr) && (F = function(t) {
  var r = $(t), n = r == Si ? t.constructor : void 0, e = n ? L(n) : "";
  if (e)
    switch (e) {
      case xi:
        return fr;
      case Ti:
        return ar;
      case Ei:
        return cr;
      case Ci:
        return ur;
      case Ii:
        return sr;
    }
  return r;
});
const Q = F;
var Pi = Object.prototype, ki = Pi.hasOwnProperty;
function Fi(t) {
  var r = t.length, n = new t.constructor(r);
  return r && typeof t[0] == "string" && ki.call(t, "index") && (n.index = t.index, n.input = t.input), n;
}
var zi = T.Uint8Array;
const st = zi;
function Rt(t) {
  var r = new t.constructor(t.byteLength);
  return new st(r).set(new st(t)), r;
}
function $i(t, r) {
  var n = r ? Rt(t.buffer) : t.buffer;
  return new t.constructor(n, t.byteOffset, t.byteLength);
}
var Li = /\w*$/;
function Bi(t) {
  var r = new t.constructor(t.source, Li.exec(t));
  return r.lastIndex = t.lastIndex, r;
}
var lr = S ? S.prototype : void 0, pr = lr ? lr.valueOf : void 0;
function Ui(t) {
  return pr ? Object(pr.call(t)) : {};
}
function Gr(t, r) {
  var n = r ? Rt(t.buffer) : t.buffer;
  return new t.constructor(n, t.byteOffset, t.length);
}
var Mi = "[object Boolean]", Di = "[object Date]", Ni = "[object Map]", Ri = "[object Number]", Wi = "[object RegExp]", Vi = "[object Set]", qi = "[object String]", Gi = "[object Symbol]", Ki = "[object ArrayBuffer]", Ji = "[object DataView]", Hi = "[object Float32Array]", Xi = "[object Float64Array]", Qi = "[object Int8Array]", Zi = "[object Int16Array]", Yi = "[object Int32Array]", ta = "[object Uint8Array]", ra = "[object Uint8ClampedArray]", na = "[object Uint16Array]", ea = "[object Uint32Array]";
function oa(t, r, n) {
  var e = t.constructor;
  switch (r) {
    case Ki:
      return Rt(t);
    case Mi:
    case Di:
      return new e(+t);
    case Ji:
      return $i(t, n);
    case Hi:
    case Xi:
    case Qi:
    case Zi:
    case Yi:
    case ta:
    case ra:
    case na:
    case ea:
      return Gr(t, n);
    case Ni:
      return new e();
    case Ri:
    case qi:
      return new e(t);
    case Wi:
      return Bi(t);
    case Vi:
      return new e();
    case Gi:
      return Ui(t);
  }
}
function Kr(t) {
  return typeof t.constructor == "function" && !$t(t) ? Hn(Dt(t)) : {};
}
var ia = "[object Map]";
function aa(t) {
  return x(t) && Q(t) == ia;
}
var vr = D && D.isMap, ca = vr ? Lt(vr) : aa;
const ua = ca;
var sa = "[object Set]";
function fa(t) {
  return x(t) && Q(t) == sa;
}
var dr = D && D.isSet, la = dr ? Lt(dr) : fa;
const pa = la;
var va = 1, da = 2, ba = 4, Jr = "[object Arguments]", ya = "[object Array]", ha = "[object Boolean]", ja = "[object Date]", ga = "[object Error]", Hr = "[object Function]", ma = "[object GeneratorFunction]", _a = "[object Map]", Oa = "[object Number]", Xr = "[object Object]", wa = "[object RegExp]", Aa = "[object Set]", Sa = "[object String]", xa = "[object Symbol]", Ta = "[object WeakMap]", Ea = "[object ArrayBuffer]", Ca = "[object DataView]", Ia = "[object Float32Array]", Pa = "[object Float64Array]", ka = "[object Int8Array]", Fa = "[object Int16Array]", za = "[object Int32Array]", $a = "[object Uint8Array]", La = "[object Uint8ClampedArray]", Ba = "[object Uint16Array]", Ua = "[object Uint32Array]", h = {};
h[Jr] = h[ya] = h[Ea] = h[Ca] = h[ha] = h[ja] = h[Ia] = h[Pa] = h[ka] = h[Fa] = h[za] = h[_a] = h[Oa] = h[Xr] = h[wa] = h[Aa] = h[Sa] = h[xa] = h[$a] = h[La] = h[Ba] = h[Ua] = !0;
h[ga] = h[Hr] = h[Ta] = !1;
function it(t, r, n, e, o, i) {
  var a, c = r & va, u = r & da, f = r & ba;
  if (n && (a = o ? n(t, e, o, i) : n(t)), a !== void 0)
    return a;
  if (!E(t))
    return t;
  var d = O(t);
  if (d) {
    if (a = Fi(t), !c)
      return Er(t, a);
  } else {
    var s = Q(t), l = s == Hr || s == ma;
    if (J(t))
      return Rr(t, c);
    if (s == Xr || s == Jr || l && !o) {
      if (a = u || l ? {} : Kr(t), !c)
        return u ? mi(t, li(a, t)) : hi(t, fi(a, t));
    } else {
      if (!h[s])
        return o ? t : {};
      a = oa(t, s, c);
    }
  }
  i || (i = new w());
  var y = i.get(t);
  if (y)
    return y;
  i.set(t, a), pa(t) ? t.forEach(function(b) {
    a.add(it(b, r, n, b, t, i));
  }) : ua(t) && t.forEach(function(b, v) {
    a.set(v, it(b, r, n, v, t, i));
  });
  var g = f ? u ? _i : Ot : u ? R : Y, p = d ? void 0 : g(t);
  return ce(p || t, function(b, v) {
    p && (v = b, b = t[v]), Cr(a, v, it(b, r, n, v, t, i));
  }), a;
}
var Ma = 1, Da = 4;
function Na(t) {
  return it(t, Ma | Da);
}
var Ra = "__lodash_hash_undefined__";
function Wa(t) {
  return this.__data__.set(t, Ra), this;
}
function Va(t) {
  return this.__data__.has(t);
}
function ft(t) {
  var r = -1, n = t == null ? 0 : t.length;
  for (this.__data__ = new I(); ++r < n; )
    this.add(t[r]);
}
ft.prototype.add = ft.prototype.push = Wa;
ft.prototype.has = Va;
function qa(t, r) {
  for (var n = -1, e = t == null ? 0 : t.length; ++n < e; )
    if (r(t[n], n, t))
      return !0;
  return !1;
}
function Ga(t, r) {
  return t.has(r);
}
var Ka = 1, Ja = 2;
function Qr(t, r, n, e, o, i) {
  var a = n & Ka, c = t.length, u = r.length;
  if (c != u && !(a && u > c))
    return !1;
  var f = i.get(t), d = i.get(r);
  if (f && d)
    return f == r && d == t;
  var s = -1, l = !0, y = n & Ja ? new ft() : void 0;
  for (i.set(t, r), i.set(r, t); ++s < c; ) {
    var g = t[s], p = r[s];
    if (e)
      var b = a ? e(p, g, s, r, t, i) : e(g, p, s, t, r, i);
    if (b !== void 0) {
      if (b)
        continue;
      l = !1;
      break;
    }
    if (y) {
      if (!qa(r, function(v, m) {
        if (!Ga(y, m) && (g === v || o(g, v, n, e, i)))
          return y.push(m);
      })) {
        l = !1;
        break;
      }
    } else if (!(g === p || o(g, p, n, e, i))) {
      l = !1;
      break;
    }
  }
  return i.delete(t), i.delete(r), l;
}
function Ha(t) {
  var r = -1, n = Array(t.size);
  return t.forEach(function(e, o) {
    n[++r] = [o, e];
  }), n;
}
function Xa(t) {
  var r = -1, n = Array(t.size);
  return t.forEach(function(e) {
    n[++r] = e;
  }), n;
}
var Qa = 1, Za = 2, Ya = "[object Boolean]", tc = "[object Date]", rc = "[object Error]", nc = "[object Map]", ec = "[object Number]", oc = "[object RegExp]", ic = "[object Set]", ac = "[object String]", cc = "[object Symbol]", uc = "[object ArrayBuffer]", sc = "[object DataView]", br = S ? S.prototype : void 0, gt = br ? br.valueOf : void 0;
function fc(t, r, n, e, o, i, a) {
  switch (n) {
    case sc:
      if (t.byteLength != r.byteLength || t.byteOffset != r.byteOffset)
        return !1;
      t = t.buffer, r = r.buffer;
    case uc:
      return !(t.byteLength != r.byteLength || !i(new st(t), new st(r)));
    case Ya:
    case tc:
    case ec:
      return Z(+t, +r);
    case rc:
      return t.name == r.name && t.message == r.message;
    case oc:
    case ac:
      return t == r + "";
    case nc:
      var c = Ha;
    case ic:
      var u = e & Qa;
      if (c || (c = Xa), t.size != r.size && !u)
        return !1;
      var f = a.get(t);
      if (f)
        return f == r;
      e |= Za, a.set(t, r);
      var d = Qr(c(t), c(r), e, o, i, a);
      return a.delete(t), d;
    case cc:
      if (gt)
        return gt.call(t) == gt.call(r);
  }
  return !1;
}
var lc = 1, pc = Object.prototype, vc = pc.hasOwnProperty;
function dc(t, r, n, e, o, i) {
  var a = n & lc, c = Ot(t), u = c.length, f = Ot(r), d = f.length;
  if (u != d && !a)
    return !1;
  for (var s = u; s--; ) {
    var l = c[s];
    if (!(a ? l in r : vc.call(r, l)))
      return !1;
  }
  var y = i.get(t), g = i.get(r);
  if (y && g)
    return y == r && g == t;
  var p = !0;
  i.set(t, r), i.set(r, t);
  for (var b = a; ++s < u; ) {
    l = c[s];
    var v = t[l], m = r[l];
    if (e)
      var U = a ? e(m, v, l, r, t, i) : e(v, m, l, t, r, i);
    if (!(U === void 0 ? v === m || o(v, m, n, e, i) : U)) {
      p = !1;
      break;
    }
    b || (b = l == "constructor");
  }
  if (p && !b) {
    var k = t.constructor, A = r.constructor;
    k != A && "constructor" in t && "constructor" in r && !(typeof k == "function" && k instanceof k && typeof A == "function" && A instanceof A) && (p = !1);
  }
  return i.delete(t), i.delete(r), p;
}
var bc = 1, yr = "[object Arguments]", hr = "[object Array]", rt = "[object Object]", yc = Object.prototype, jr = yc.hasOwnProperty;
function hc(t, r, n, e, o, i) {
  var a = O(t), c = O(r), u = a ? hr : Q(t), f = c ? hr : Q(r);
  u = u == yr ? rt : u, f = f == yr ? rt : f;
  var d = u == rt, s = f == rt, l = u == f;
  if (l && J(t)) {
    if (!J(r))
      return !1;
    a = !0, d = !1;
  }
  if (l && !d)
    return i || (i = new w()), a || Bt(t) ? Qr(t, r, n, e, o, i) : fc(t, r, u, n, e, o, i);
  if (!(n & bc)) {
    var y = d && jr.call(t, "__wrapped__"), g = s && jr.call(r, "__wrapped__");
    if (y || g) {
      var p = y ? t.value() : t, b = g ? r.value() : r;
      return i || (i = new w()), o(p, b, n, e, i);
    }
  }
  return l ? (i || (i = new w()), dc(t, r, n, e, o, i)) : !1;
}
function Wt(t, r, n, e, o) {
  return t === r ? !0 : t == null || r == null || !x(t) && !x(r) ? t !== t && r !== r : hc(t, r, n, e, Wt, o);
}
var jc = 1, gc = 2;
function mc(t, r, n, e) {
  var o = n.length, i = o, a = !e;
  if (t == null)
    return !i;
  for (t = Object(t); o--; ) {
    var c = n[o];
    if (a && c[2] ? c[1] !== t[c[0]] : !(c[0] in t))
      return !1;
  }
  for (; ++o < i; ) {
    c = n[o];
    var u = c[0], f = t[u], d = c[1];
    if (a && c[2]) {
      if (f === void 0 && !(u in t))
        return !1;
    } else {
      var s = new w();
      if (e)
        var l = e(f, d, u, t, r, s);
      if (!(l === void 0 ? Wt(d, f, jc | gc, e, s) : l))
        return !1;
    }
  }
  return !0;
}
function Zr(t) {
  return t === t && !E(t);
}
function _c(t) {
  for (var r = Y(t), n = r.length; n--; ) {
    var e = r[n], o = t[e];
    r[n] = [e, o, Zr(o)];
  }
  return r;
}
function Yr(t, r) {
  return function(n) {
    return n == null ? !1 : n[t] === r && (r !== void 0 || t in Object(n));
  };
}
function Oc(t) {
  var r = _c(t);
  return r.length == 1 && r[0][2] ? Yr(r[0][0], r[0][1]) : function(n) {
    return n === t || mc(n, t, r);
  };
}
function wc(t, r) {
  return t != null && r in Object(t);
}
function Ac(t, r, n) {
  r = Lr(r, t);
  for (var e = -1, o = r.length, i = !1; ++e < o; ) {
    var a = yt(r[e]);
    if (!(i = t != null && n(t, a)))
      break;
    t = t[a];
  }
  return i || ++e != o ? i : (o = t == null ? 0 : t.length, !!o && zt(o) && Ft(a, o) && (O(t) || ut(t)));
}
function Sc(t, r) {
  return t != null && Ac(t, r, wc);
}
var xc = 1, Tc = 2;
function Ec(t, r) {
  return Ut(t) && Zr(r) ? Yr(yt(t), r) : function(n) {
    var e = Qo(n, t);
    return e === void 0 && e === r ? Sc(n, t) : Wt(r, e, xc | Tc);
  };
}
function Cc(t) {
  return function(r) {
    return r == null ? void 0 : r[t];
  };
}
function Ic(t) {
  return function(r) {
    return Br(r, t);
  };
}
function Pc(t) {
  return Ut(t) ? Cc(yt(t)) : Ic(t);
}
function kc(t) {
  return typeof t == "function" ? t : t == null ? Pt : typeof t == "object" ? O(t) ? Ec(t[0], t[1]) : Oc(t) : Pc(t);
}
function Fc(t) {
  return function(r, n, e) {
    for (var o = -1, i = Object(r), a = e(r), c = a.length; c--; ) {
      var u = a[t ? c : ++o];
      if (n(i[u], u, i) === !1)
        break;
    }
    return r;
  };
}
var zc = Fc();
const tn = zc;
function $c(t, r) {
  return t && tn(t, r, Y);
}
function xt(t, r, n) {
  (n !== void 0 && !Z(t[r], n) || n === void 0 && !(r in t)) && pt(t, r, n);
}
function Lc(t) {
  return x(t) && vt(t);
}
function Tt(t, r) {
  if (!(r === "constructor" && typeof t[r] == "function") && r != "__proto__")
    return t[r];
}
function Bc(t) {
  return N(t, R(t));
}
function Uc(t, r, n, e, o, i, a) {
  var c = Tt(t, n), u = Tt(r, n), f = a.get(u);
  if (f) {
    xt(t, n, f);
    return;
  }
  var d = i ? i(c, u, n + "", t, r, a) : void 0, s = d === void 0;
  if (s) {
    var l = O(u), y = !l && J(u), g = !l && !y && Bt(u);
    d = u, l || y || g ? O(c) ? d = c : Lc(c) ? d = Er(c) : y ? (s = !1, d = Rr(u, !0)) : g ? (s = !1, d = Gr(u, !0)) : d = [] : Dr(u) || ut(u) ? (d = c, ut(c) ? d = Bc(c) : (!E(c) || kt(c)) && (d = Kr(u))) : s = !1;
  }
  s && (a.set(u, d), o(d, u, e, i, a), a.delete(u)), xt(t, n, d);
}
function rn(t, r, n, e, o) {
  t !== r && tn(r, function(i, a) {
    if (o || (o = new w()), E(i))
      Uc(t, r, a, n, rn, e, o);
    else {
      var c = e ? e(Tt(t, a), i, a + "", t, r, o) : void 0;
      c === void 0 && (c = i), xt(t, a, c);
    }
  }, R);
}
var Mc = Ir(function(t, r, n, e) {
  rn(t, r, n, e);
});
const Dc = Mc;
function Nc(t, r) {
  var n = {};
  return r = kc(r), $c(t, function(e, o, i) {
    pt(n, r(e, o, i), e);
  }), n;
}
const Rc = new RegExp("([\\p{Ll}\\d])(\\p{Lu})", "gu"), Wc = new RegExp("(\\p{Lu})([\\p{Lu}][\\p{Ll}])", "gu"), Vc = new RegExp("(\\d)(\\p{Ll})", "gu"), qc = new RegExp("(\\p{L})(\\d)", "gu"), Gc = /[^\p{L}\d]+/giu, nt = "$1\0$2", gr = "";
function Kc(t, r) {
  let n = t.trim();
  n = n.replace(Rc, nt).replace(Wc, nt), r != null && r.separateNumbers && (n = n.replace(Vc, nt).replace(qc, nt)), n = n.replace(Gc, "\0");
  let e = 0, o = n.length;
  for (; n.charAt(e) === "\0"; )
    e++;
  if (e === o)
    return [];
  for (; n.charAt(o - 1) === "\0"; )
    o--;
  return n.slice(e, o).split(/\0/g);
}
function Jc(t, r) {
  const [n, e, o] = Yc(t, r), i = Hc(r == null ? void 0 : r.locale), a = Xc(r == null ? void 0 : r.locale), c = r != null && r.mergeAmbiguousCharacters ? Qc(i, a) : Zc(i, a);
  return n + Kc(e, r).map((u, f) => f === 0 ? i(u) : c(u, f)).join((r == null ? void 0 : r.delimiter) ?? "") + o;
}
function Hc(t) {
  return t === !1 ? (r) => r.toLowerCase() : (r) => r.toLocaleLowerCase(t);
}
function Xc(t) {
  return t === !1 ? (r) => r.toUpperCase() : (r) => r.toLocaleUpperCase(t);
}
function Qc(t, r) {
  return (n) => `${r(n[0])}${t(n.slice(1))}`;
}
function Zc(t, r) {
  return (n, e) => {
    const o = n[0];
    return (e > 0 && o >= "0" && o <= "9" ? "_" + o : r(o)) + t(n.slice(1));
  };
}
function Yc(t, r) {
  const n = (r == null ? void 0 : r.prefixCharacters) ?? gr, e = (r == null ? void 0 : r.suffixCharacters) ?? gr;
  let o = 0, i = t.length;
  for (; o < t.length; ) {
    const a = t.charAt(o);
    if (!n.includes(a))
      break;
    o++;
  }
  for (; i > o; ) {
    const a = i - 1, c = t.charAt(a);
    if (!e.includes(c))
      break;
    i = a;
  }
  return [
    t.slice(0, o),
    t.slice(o, i),
    t.slice(i)
  ];
}
const K = (t) => /.+-.+/.test(t) ? Jc(t) : t;
function tu(t, r) {
  const n = /* @__PURE__ */ Object.create(null), e = t.split(",");
  for (let o = 0; o < e.length; o++)
    n[e[o]] = !0;
  return r ? (o) => !!n[o.toLowerCase()] : (o) => !!n[o];
}
const ru = /* @__PURE__ */ tu("String,Number,Boolean,Function,Symbol,BigInt");
function nu(t) {
  const r = t == null ? void 0 : t.toString().match(/^\s*function (\w+)/);
  return r ? r[1] : t === null ? "null" : "";
}
function eu(t, r) {
  let n;
  const e = nu(r);
  if (ru(e)) {
    const o = typeof t;
    n = o === e.toLowerCase(), !n && o === "object" && (n = t instanceof r);
  } else
    e === "Object" ? n = E(t) : e === "Array" ? n = Array.isArray(t) : e === "null" ? n = t === null : n = t instanceof r;
  return {
    valid: n,
    expectedType: e
  };
}
function ou({
  prop: t,
  type: r,
  validator: n
}) {
  if (![void 0, null].includes(t) && r) {
    let e = !1;
    const o = Array.isArray(r) ? r : [r], i = [];
    for (let a = 0; a < o.length && !e; a++) {
      const { valid: c, expectedType: u } = eu(t, o[a]);
      i.push(u || ""), e = c;
    }
    if (!e)
      throw new TypeError(
        `Invalid prop: type check failed, expecting [${i.join(
          ", "
        )}], receiving: ${t}`
      );
  }
  if (n && !n(t))
    throw new Error(`Invalid prop: validator check failed, receiving: ${t}`);
}
function iu(t, {
  mergeObject: r,
  mergeFunction: n
}) {
  const e = [];
  for (let i = t.length - 1; i >= 0; i--)
    e.push(t[i]);
  const o = n ? (i, a) => typeof i == "function" && typeof a == "function" ? n(a, i) : void 0 : void 0;
  return r === "deep" ? Dc(...e, o) : po(...e, o);
}
function au(t, {
  mergeFunction: r
}) {
  return t.reduce(r, () => {
  });
}
function M(t, r = {}) {
  const {
    type: n,
    default: e,
    defaultIsDynamic: o = !1,
    required: i = !1,
    validator: a,
    camelizeObjectKeys: c = !1,
    mergeObjectApplyOnlyToDefault: u = !1,
    mergeFunctionApplyOnlyToDefault: f = !0
  } = r;
  let { mergeObject: d = "deep", mergeFunction: s = !1 } = r;
  const l = [];
  let y, g = !1, p = !1;
  const b = (v) => {
    if (v !== void 0) {
      ou({ type: n, prop: v, validator: a });
      const m = Dr(v), U = typeof v == "function";
      return g = m, p = U, m ? (v = Na(v), c ? Nc(v, (k, A) => K(A)) : v) : v;
    }
  };
  for (const v of t)
    l.push(b(v));
  if (!o)
    l.push(b(e));
  else if (typeof e != "function")
    throw new TypeError(
      `Invalid option: options.default should be Function when options.defaultIsDynamic enabled, receiving: ${e}`
    );
  g || (d = !1, p || (s = !1));
  for (let v = 0; v < l.length; v++) {
    const m = l[v];
    if (m !== void 0) {
      v === l.length - 1 ? y = m : d ? y = iu(
        u ? [m, e] : l,
        {
          mergeObject: d,
          mergeFunction: s
        }
      ) : s ? y = au(
        f ? [m, e] : l,
        {
          mergeFunction: s
        }
      ) : y = m;
      break;
    }
  }
  if (i && [void 0, null].includes(y))
    throw new Error("Missing required prop");
  return o ? M(t, {
    ...r,
    default: e(y),
    defaultIsDynamic: !1
  }) : y;
}
function mr(t) {
  const r = Array.from(K(t));
  return r[0] = r[0].toUpperCase(), r.unshift("o", "n"), r.join("");
}
function cu(t, {
  props: r = [],
  camelizePropNames: n = !1
} = {}) {
  const e = {
    props: {},
    attrs: {},
    listeners: {},
    hooks: {},
    slots: {}
  };
  let o;
  if (Array.isArray(r))
    o = n ? r.map((i) => K(i)) : r;
  else if (n) {
    o = [];
    for (const i in r)
      o.push(K(i));
  } else
    o = Object.keys(r);
  for (const i in t)
    if (i.startsWith("@")) {
      const a = i.substring(1);
      if (Ct)
        if (a.startsWith("vue:"))
          e.hooks[mr(a.replace("vue:", "vnode-"))] = t[i];
        else {
          if (a.startsWith("vnode"))
            throw new Error("@vnode-* hooks are no longer supported. Use the vue: prefix instead. For example, @vnode-mounted should be changed to @vue:mounted. @vnode-* hooks support has been removed in Vue 3.4.");
          e.listeners[mr(a)] = t[i];
        }
      else
        a.startsWith("hook:") ? e.hooks[a] = t[i] : e.listeners[a] = t[i];
    } else if (i.startsWith("#")) {
      const a = i.substring(1);
      a && (e.slots[a] = t[i]);
    } else {
      const a = n ? K(i) : i;
      o.includes(a) ? e.props[a] = t[i] : e.attrs[i] = t[i];
    }
  return e;
}
var uu = typeof global == "object" && global && global.Object === Object && global, su = typeof self == "object" && self && self.Object === Object && self, nn = uu || su || Function("return this")(), lt = nn.Symbol, en = Object.prototype, fu = en.hasOwnProperty, lu = en.toString, q = lt ? lt.toStringTag : void 0;
function pu(t) {
  var r = fu.call(t, q), n = t[q];
  try {
    t[q] = void 0;
    var e = !0;
  } catch {
  }
  var o = lu.call(t);
  return e && (r ? t[q] = n : delete t[q]), o;
}
var vu = Object.prototype, du = vu.toString;
function bu(t) {
  return du.call(t);
}
var yu = "[object Null]", hu = "[object Undefined]", _r = lt ? lt.toStringTag : void 0;
function ju(t) {
  return t == null ? t === void 0 ? hu : yu : _r && _r in Object(t) ? pu(t) : bu(t);
}
function gu(t) {
  return t != null && typeof t == "object";
}
var mu = "[object Symbol]";
function _u(t) {
  return typeof t == "symbol" || gu(t) && ju(t) == mu;
}
var Ou = /\s/;
function wu(t) {
  for (var r = t.length; r-- && Ou.test(t.charAt(r)); )
    ;
  return r;
}
var Au = /^\s+/;
function Su(t) {
  return t && t.slice(0, wu(t) + 1).replace(Au, "");
}
function Et(t) {
  var r = typeof t;
  return t != null && (r == "object" || r == "function");
}
var Or = NaN, xu = /^[-+]0x[0-9a-f]+$/i, Tu = /^0b[01]+$/i, Eu = /^0o[0-7]+$/i, Cu = parseInt;
function wr(t) {
  if (typeof t == "number")
    return t;
  if (_u(t))
    return Or;
  if (Et(t)) {
    var r = typeof t.valueOf == "function" ? t.valueOf() : t;
    t = Et(r) ? r + "" : r;
  }
  if (typeof t != "string")
    return t === 0 ? t : +t;
  t = Su(t);
  var n = Tu.test(t);
  return n || Eu.test(t) ? Cu(t.slice(2), n ? 2 : 8) : xu.test(t) ? Or : +t;
}
var mt = function() {
  return nn.Date.now();
}, Iu = "Expected a function", Pu = Math.max, ku = Math.min;
function Fu(t, r, n) {
  var e, o, i, a, c, u, f = 0, d = !1, s = !1, l = !0;
  if (typeof t != "function")
    throw new TypeError(Iu);
  r = wr(r) || 0, Et(n) && (d = !!n.leading, s = "maxWait" in n, i = s ? Pu(wr(n.maxWait) || 0, r) : i, l = "trailing" in n ? !!n.trailing : l);
  function y(_) {
    var P = e, W = o;
    return e = o = void 0, f = _, a = t.apply(W, P), a;
  }
  function g(_) {
    return f = _, c = setTimeout(v, r), d ? y(_) : a;
  }
  function p(_) {
    var P = _ - u, W = _ - f, Vt = r - P;
    return s ? ku(Vt, i - W) : Vt;
  }
  function b(_) {
    var P = _ - u, W = _ - f;
    return u === void 0 || P >= r || P < 0 || s && W >= i;
  }
  function v() {
    var _ = mt();
    if (b(_))
      return m(_);
    c = setTimeout(v, p(_));
  }
  function m(_) {
    return c = void 0, l && e ? y(_) : (e = o = void 0, a);
  }
  function U() {
    c !== void 0 && clearTimeout(c), f = 0, e = u = o = c = void 0;
  }
  function k() {
    return c === void 0 ? a : m(mt());
  }
  function A() {
    var _ = mt(), P = b(_);
    if (e = arguments, o = this, u = _, P) {
      if (c === void 0)
        return g(u);
      if (s)
        return clearTimeout(c), c = setTimeout(v, r), y(u);
    }
    return c === void 0 && (c = setTimeout(v, r)), a;
  }
  return A.cancel = U, A.flush = k, A;
}
const zu = "JsonEditorVue", et = Ct ? "modelValue" : "value", Ar = Ct ? "update:modelValue" : "input", ot = [
  "mainMenuBar",
  "navigationBar",
  "statusBar",
  "askToFormat",
  "readOnly",
  "escapeControlCharacters",
  "escapeUnicodeCharacters",
  "flattenColumns"
], on = cn({
  name: zu,
  props: {
    [et]: {},
    mode: {
      type: String
    },
    ...Object.fromEntries(
      ot.map((t) => [
        t,
        {
          type: Boolean,
          default: void 0
        }
      ])
    )
  },
  emits: {
    [Ar](t) {
      return !0;
    },
    "update:mode": function(t) {
      return !0;
    }
  },
  setup(t, { attrs: r, emit: n, expose: e }) {
    var d;
    const o = (d = un()) == null ? void 0 : d.proxy, i = qt(), a = qt(!1), c = Fu((s) => {
      a.value = !0, n(
        Ar,
        s.text === void 0 ? s.json : s.text
      );
    }, 100), u = (s) => {
      n("update:mode", s);
    }, f = (s, l) => (...y) => {
      s(...y), l(...y);
    };
    return e == null || e({ jsonEditor: i }), sn(() => {
      var s;
      (s = i.value) == null || s.destroy();
    }), fn(() => {
      const s = M([t.mode, at.mode], {
        type: String
      }), l = M([t[et], at[et]]), y = Object.fromEntries(
        Array.from(ot, (p) => [p, M([t[p], at[p]])]).filter(
          ([, p]) => p !== void 0
        )
      ), g = M(
        [
          y,
          r,
          an,
          {
            onChange: c,
            onChangeMode: u,
            mode: s,
            ...l !== void 0 && {
              content: typeof l == "string" ? { text: l } : { json: l }
            }
          }
        ],
        {
          camelizeObjectKeys: !0,
          mergeFunction: f,
          type: Object
        }
      );
      i.value = new vn({
        target: o == null ? void 0 : o.$refs.jsonEditorRef,
        props: g
      }), tt(
        () => t[et],
        (p) => {
          if (a.value) {
            a.value = !1;
            return;
          }
          i.value && i.value.set(
            [void 0, ""].includes(p) ? { text: "" } : typeof p == "string" ? { text: p } : { json: p }
          );
        },
        {
          deep: !0
        }
      ), tt(
        () => t.mode,
        (p) => {
          var b;
          (b = i.value) == null || b.updateProps({
            mode: p
          });
        }
      ), tt(
        () => Array.from(ot, (p) => t[p]),
        (p) => {
          var b;
          (b = i.value) == null || b.updateProps(
            Object.fromEntries(Array.from(p, (v, m) => [ot[m], v]).filter(([, v]) => v !== void 0))
          );
        }
      ), tt(
        () => r,
        (p) => {
          var v;
          const b = {};
          (p.onChange || p["on-change"]) && (b.onChange = c), (p.onChangeMode || p["on-change-mode"]) && (b.onChangeMode = u), (v = i.value) == null || v.updateProps(
            Object.getOwnPropertyNames(b).length > 0 ? M([p, b], {
              camelizeObjectKeys: !0,
              mergeFunction: f,
              type: Object
            }) : p
          );
        },
        {
          deep: !0
        }
      ), e || (e = (p) => {
        for (const b in p)
          o[b] = ln(p[b]);
      }, e({ jsonEditor: i }));
    }), () => pn("div", { ref: "jsonEditorRef" });
  }
}), at = {}, an = {};
function $u(t) {
  return t.install = (r, n = {}) => {
    const { props: e, attrs: o } = cu(n, { props: on.props });
    Object.assign(at, e), Object.assign(an, o), r.component(t.name, t);
  }, t;
}
const Uu = $u(on);
export {
  Uu as default
};
