// @vue/devtools-applet/dist/index.js
import { onRpcConnected as Re, rpc as D, DevToolsMessagingEvents as W, useDevToolsState as Jt } from "@vue/devtools-core";
import { ref as T, onUnmounted as ye, isRef as Ge, unref as d, watch as X, onMounted as St, nextTick as ze, readonly as Yt, getCurrentScope as Xt, onScopeDispose as Qt, getCurrentInstance as Qe, computed as V, reactive as ft, shallowRef as vt, toRaw as Tt, defineComponent as B, withDirectives as J, openBlock as p, createElementBlock as m, normalizeStyle as Ie, createTextVNode as ne, toDisplayString as le, mergeModels as Be, useModel as et, Fragment as te, renderList as ae, normalizeClass as ie, createBlock as z, provide as ge, inject as ve, mergeProps as ke, withModifiers as fe, withCtx as M, createVNode as x, createCommentVNode as U, createElementVNode as r, watchEffect as tt, renderSlot as nt, toValue as mt, h as en, vShow as tn, createStaticVNode as nn } from "vue";
import { isPlainObject as on, UNDEFINED as ln, NAN as sn, INFINITY as an, NEGATIVE_INFINITY as rn, getRaw as Et, toSubmit as Ue, formatInspectorStateValue as un, getInspectorStateValueType as ht, toEdit as cn, parse as Y } from "@vue/devtools-kit";
import { isArray as gt, isObject as Ct, sortByKey as Vt, isInSeparateWindow as dn, isInChromePanel as _t } from "@vue/devtools-shared";
import { vTooltip as Q, VueButton as de, VueIcon as oe, VueDropdown as pn, VueDropdownButton as xt, VueInput as Ze, VueCodeBlock as fn, VueDialog as vn, VueSwitch as mn, VueSelect as hn, VueIcIcon as yt } from "@vue/devtools-ui";
import { flatten as gn, groupBy as Cn } from "lodash-es";
import { Splitpanes as Ae, Pane as ue } from "splitpanes";
import { debounce as _n } from "perfect-debounce";
import { RecycleScroller as xn } from "vue-virtual-scroller";
function Wl() {
  const n = T([]), e = T([]);
  function o(s) {
    e.value = s, s.length < n.value.length && (n.value = []), e.value.forEach((l) => {
      t(l);
    });
  }
  Re(() => {
    D.value.getCustomInspector().then((s) => {
      e.value = s, e.value.forEach((l) => {
        t(l);
      });
    }), D.functions.on(W.INSPECTOR_UPDATED, o);
  });
  function t(s) {
    n.value.some((l) => l.id === s.id) || n.value.push(s);
  }
  return ye(() => {
    D.functions.off(W.INSPECTOR_UPDATED, o);
  }), {
    registeredInspector: n,
    register: t
  };
}
function ot(n) {
  return Xt() ? (Qt(n), !0) : !1;
}
function ee(n) {
  return typeof n == "function" ? n() : d(n);
}
const lt = typeof window < "u" && typeof document < "u";
typeof WorkerGlobalScope < "u" && globalThis instanceof WorkerGlobalScope;
const yn = (n) => typeof n < "u", bn = Object.prototype.toString, wn = (n) => bn.call(n) === "[object Object]", Fe = () => {
};
function kn(n, e) {
  function o(...t) {
    return new Promise((s, l) => {
      Promise.resolve(n(() => e.apply(this, t), { fn: e, thisArg: this, args: t })).then(s).catch(l);
    });
  }
  return o;
}
const In = (n) => n();
function $n(n, e = {}) {
  let o, t, s = Fe;
  const l = (u) => {
    clearTimeout(u), s(), s = Fe;
  };
  return (u) => {
    const c = ee(n), C = ee(e.maxWait);
    return o && l(o), c <= 0 || C !== void 0 && C <= 0 ? (t && (l(t), t = null), Promise.resolve(u())) : new Promise((v, a) => {
      s = e.rejectOnCancel ? a : v, C && !t && (t = setTimeout(() => {
        o && l(o), t = null, v(u());
      }, C)), o = setTimeout(() => {
        t && l(t), t = null, v(u());
      }, c);
    });
  };
}
function bt(n, e = !1, o = "Timeout") {
  return new Promise((t, s) => {
    setTimeout(e ? () => s(o) : t, n);
  });
}
function Sn(n) {
  let e;
  function o() {
    return e || (e = n()), e;
  }
  return o.reset = async () => {
    const t = e;
    e = void 0, t && await t;
  }, o;
}
function Tn(n) {
  return Qe();
}
function En(n, e, o = {}) {
  const {
    eventFilter: t = In,
    ...s
  } = o;
  return X(
    n,
    kn(
      t,
      e
    ),
    s
  );
}
function Vn(n, e = !0, o) {
  Tn() ? St(n, o) : e ? n() : ze(n);
}
function Xe(n, e = !1) {
  function o(a, { flush: f = "sync", deep: g = !1, timeout: b, throwOnTimeout: h } = {}) {
    let S = null;
    const O = [new Promise((A) => {
      S = X(
        n,
        (k) => {
          a(k) !== e && (S ? S() : ze(() => S == null ? void 0 : S()), A(k));
        },
        {
          flush: f,
          deep: g,
          immediate: !0
        }
      );
    })];
    return b != null && O.push(
      bt(b, h).then(() => ee(n)).finally(() => S == null ? void 0 : S())
    ), Promise.race(O);
  }
  function t(a, f) {
    if (!Ge(a))
      return o((k) => k === a, f);
    const { flush: g = "sync", deep: b = !1, timeout: h, throwOnTimeout: S } = f ?? {};
    let j = null;
    const A = [new Promise((k) => {
      j = X(
        [n, a],
        ([R, H]) => {
          e !== (R === H) && (j ? j() : ze(() => j == null ? void 0 : j()), k(R));
        },
        {
          flush: g,
          deep: b,
          immediate: !0
        }
      );
    })];
    return h != null && A.push(
      bt(h, S).then(() => ee(n)).finally(() => (j == null || j(), ee(n)))
    ), Promise.race(A);
  }
  function s(a) {
    return o((f) => !!f, a);
  }
  function l(a) {
    return t(null, a);
  }
  function i(a) {
    return t(void 0, a);
  }
  function u(a) {
    return o(Number.isNaN, a);
  }
  function c(a, f) {
    return o((g) => {
      const b = Array.from(g);
      return b.includes(a) || b.includes(ee(a));
    }, f);
  }
  function C(a) {
    return v(1, a);
  }
  function v(a = 1, f) {
    let g = -1;
    return o(() => (g += 1, g >= a), f);
  }
  return Array.isArray(ee(n)) ? {
    toMatch: o,
    toContains: c,
    changed: C,
    changedTimes: v,
    get not() {
      return Xe(n, !e);
    }
  } : {
    toMatch: o,
    toBe: t,
    toBeTruthy: s,
    toBeNull: l,
    toBeNaN: u,
    toBeUndefined: i,
    changed: C,
    changedTimes: v,
    get not() {
      return Xe(n, !e);
    }
  };
}
function Nt(n) {
  return Xe(n);
}
function Nn(n, e, o = {}) {
  const {
    immediate: t = !0
  } = o, s = T(!1);
  let l = null;
  function i() {
    l && (clearTimeout(l), l = null);
  }
  function u() {
    s.value = !1, i();
  }
  function c(...C) {
    i(), s.value = !0, l = setTimeout(() => {
      s.value = !1, l = null, n(...C);
    }, ee(e));
  }
  return t && (s.value = !0, lt && c()), ot(u), {
    isPending: Yt(s),
    start: c,
    stop: u
  };
}
function Rn(n = !1, e = {}) {
  const {
    truthyValue: o = !0,
    falsyValue: t = !1
  } = e, s = Ge(n), l = T(n);
  function i(u) {
    if (arguments.length)
      return l.value = u, l.value;
    {
      const c = ee(o);
      return l.value = l.value === c ? ee(t) : c, l.value;
    }
  }
  return s ? i : [l, i];
}
function An(n, e, o = {}) {
  const {
    debounce: t = 0,
    maxWait: s = void 0,
    ...l
  } = o;
  return En(
    n,
    e,
    {
      ...l,
      eventFilter: $n(t, { maxWait: s })
    }
  );
}
const Ke = lt ? window : void 0, Rt = lt ? window.navigator : void 0;
function xe(n) {
  var e;
  const o = ee(n);
  return (e = o == null ? void 0 : o.$el) != null ? e : o;
}
function he(...n) {
  let e, o, t, s;
  if (typeof n[0] == "string" || Array.isArray(n[0]) ? ([o, t, s] = n, e = Ke) : [e, o, t, s] = n, !e)
    return Fe;
  Array.isArray(o) || (o = [o]), Array.isArray(t) || (t = [t]);
  const l = [], i = () => {
    l.forEach((v) => v()), l.length = 0;
  }, u = (v, a, f, g) => (v.addEventListener(a, f, g), () => v.removeEventListener(a, f, g)), c = X(
    () => [xe(e), ee(s)],
    ([v, a]) => {
      if (i(), !v)
        return;
      const f = wn(a) ? { ...a } : a;
      l.push(
        ...o.flatMap((g) => t.map((b) => u(v, g, b, f)))
      );
    },
    { immediate: !0, flush: "post" }
  ), C = () => {
    c(), i();
  };
  return ot(C), C;
}
function Dn() {
  const n = T(!1), e = Qe();
  return e && St(() => {
    n.value = !0;
  }, e), n;
}
function st(n) {
  const e = Dn();
  return V(() => (e.value, !!n()));
}
function wt(n, e = {}) {
  const {
    controls: o = !1,
    navigator: t = Rt
  } = e, s = st(() => t && "permissions" in t), l = vt(), i = typeof n == "string" ? { name: n } : n, u = vt(), c = () => {
    var v, a;
    u.value = (a = (v = l.value) == null ? void 0 : v.state) != null ? a : "prompt";
  };
  he(l, "change", c);
  const C = Sn(async () => {
    if (s.value) {
      if (!l.value)
        try {
          l.value = await t.permissions.query(i);
        } catch {
          l.value = void 0;
        } finally {
          c();
        }
      if (o)
        return Tt(l.value);
    }
  });
  return C(), o ? {
    state: u,
    isSupported: s,
    query: C
  } : u;
}
function Pn(n = {}) {
  const {
    navigator: e = Rt,
    read: o = !1,
    source: t,
    copiedDuring: s = 1500,
    legacy: l = !1
  } = n, i = st(() => e && "clipboard" in e), u = wt("clipboard-read"), c = wt("clipboard-write"), C = V(() => i.value || l), v = T(""), a = T(!1), f = Nn(() => a.value = !1, s);
  function g() {
    i.value && j(u.value) ? e.clipboard.readText().then((O) => {
      v.value = O;
    }) : v.value = S();
  }
  C.value && o && he(["copy", "cut"], g);
  async function b(O = ee(t)) {
    C.value && O != null && (i.value && j(c.value) ? await e.clipboard.writeText(O) : h(O), v.value = O, a.value = !0, f.start());
  }
  function h(O) {
    const A = document.createElement("textarea");
    A.value = O ?? "", A.style.position = "absolute", A.style.opacity = "0", document.body.appendChild(A), A.select(), document.execCommand("copy"), A.remove();
  }
  function S() {
    var O, A, k;
    return (k = (A = (O = document == null ? void 0 : document.getSelection) == null ? void 0 : O.call(document)) == null ? void 0 : A.toString()) != null ? k : "";
  }
  function j(O) {
    return O === "granted" || O === "prompt";
  }
  return {
    isSupported: C,
    text: v,
    copied: a,
    copy: b
  };
}
function Mn(n) {
  return JSON.parse(JSON.stringify(n));
}
function Hn(n, e, o = {}) {
  const { window: t = Ke, ...s } = o;
  let l;
  const i = st(() => t && "ResizeObserver" in t), u = () => {
    l && (l.disconnect(), l = void 0);
  }, c = V(() => {
    const a = ee(n);
    return Array.isArray(a) ? a.map((f) => xe(f)) : [xe(a)];
  }), C = X(
    c,
    (a) => {
      if (u(), i.value && t) {
        l = new ResizeObserver(e);
        for (const f of a)
          f && l.observe(f, s);
      }
    },
    { immediate: !0, flush: "post" }
  ), v = () => {
    u(), C();
  };
  return ot(v), {
    isSupported: i,
    stop: v
  };
}
function Ln(n, e = { width: 0, height: 0 }, o = {}) {
  const { window: t = Ke, box: s = "content-box" } = o, l = V(() => {
    var a, f;
    return (f = (a = xe(n)) == null ? void 0 : a.namespaceURI) == null ? void 0 : f.includes("svg");
  }), i = T(e.width), u = T(e.height), { stop: c } = Hn(
    n,
    ([a]) => {
      const f = s === "border-box" ? a.borderBoxSize : s === "content-box" ? a.contentBoxSize : a.devicePixelContentBoxSize;
      if (t && l.value) {
        const g = xe(n);
        if (g) {
          const b = g.getBoundingClientRect();
          i.value = b.width, u.value = b.height;
        }
      } else if (f) {
        const g = Array.isArray(f) ? f : [f];
        i.value = g.reduce((b, { inlineSize: h }) => b + h, 0), u.value = g.reduce((b, { blockSize: h }) => b + h, 0);
      } else
        i.value = a.contentRect.width, u.value = a.contentRect.height;
    },
    o
  );
  Vn(() => {
    const a = xe(n);
    a && (i.value = "offsetWidth" in a ? a.offsetWidth : e.width, u.value = "offsetHeight" in a ? a.offsetHeight : e.height);
  });
  const C = X(
    () => xe(n),
    (a) => {
      i.value = a ? e.width : 0, u.value = a ? e.height : 0;
    }
  );
  function v() {
    c(), C();
  }
  return {
    width: i,
    height: u,
    stop: v
  };
}
const On = {
  ctrl: "control",
  command: "meta",
  cmd: "meta",
  option: "alt",
  up: "arrowup",
  down: "arrowdown",
  left: "arrowleft",
  right: "arrowright"
};
function jn(n = {}) {
  const {
    reactive: e = !1,
    target: o = Ke,
    aliasMap: t = On,
    passive: s = !0,
    onEventFired: l = Fe
  } = n, i = ft(/* @__PURE__ */ new Set()), u = {
    toJSON() {
      return {};
    },
    current: i
  }, c = e ? ft(u) : u, C = /* @__PURE__ */ new Set(), v = /* @__PURE__ */ new Set();
  function a(h, S) {
    h in c && (e ? c[h] = S : c[h].value = S);
  }
  function f() {
    i.clear();
    for (const h of v)
      a(h, !1);
  }
  function g(h, S) {
    var j, O;
    const A = (j = h.key) == null ? void 0 : j.toLowerCase(), R = [(O = h.code) == null ? void 0 : O.toLowerCase(), A].filter(Boolean);
    A && (S ? i.add(A) : i.delete(A));
    for (const H of R)
      v.add(H), a(H, S);
    A === "meta" && !S ? (C.forEach((H) => {
      i.delete(H), a(H, !1);
    }), C.clear()) : typeof h.getModifierState == "function" && h.getModifierState("Meta") && S && [...i, ...R].forEach((H) => C.add(H));
  }
  he(o, "keydown", (h) => (g(h, !0), l(h)), { passive: s }), he(o, "keyup", (h) => (g(h, !1), l(h)), { passive: s }), he("blur", f, { passive: !0 }), he("focus", f, { passive: !0 });
  const b = new Proxy(
    c,
    {
      get(h, S, j) {
        if (typeof S != "string")
          return Reflect.get(h, S, j);
        if (S = S.toLowerCase(), S in t && (S = t[S]), !(S in c))
          if (/[+_-]/.test(S)) {
            const A = S.split(/[+_-]/g).map((k) => k.trim());
            c[S] = V(() => A.every((k) => ee(b[k])));
          } else
            c[S] = T(!1);
        const O = Reflect.get(h, S, j);
        return e ? ee(O) : O;
      }
    }
  );
  return b;
}
function Un(n, e, o, t = {}) {
  var s, l, i;
  const {
    clone: u = !1,
    passive: c = !1,
    eventName: C,
    deep: v = !1,
    defaultValue: a,
    shouldEmit: f
  } = t, g = Qe(), b = o || (g == null ? void 0 : g.emit) || ((s = g == null ? void 0 : g.$emit) == null ? void 0 : s.bind(g)) || ((i = (l = g == null ? void 0 : g.proxy) == null ? void 0 : l.$emit) == null ? void 0 : i.bind(g == null ? void 0 : g.proxy));
  let h = C;
  h = h || `update:${e.toString()}`;
  const S = (A) => u ? typeof u == "function" ? u(A) : Mn(A) : A, j = () => yn(n[e]) ? S(n[e]) : a, O = (A) => {
    f ? f(A) && b(h, A) : b(h, A);
  };
  if (c) {
    const A = j(), k = T(A);
    let R = !1;
    return X(
      () => n[e],
      (H) => {
        R || (R = !0, k.value = S(H), ze(() => R = !1));
      }
    ), X(
      k,
      (H) => {
        !R && (H !== n[e] || v) && O(H);
      },
      { deep: v }
    ), k;
  } else
    return V({
      get() {
        return j();
      },
      set(A) {
        O(A);
      }
    });
}
function kt(n) {
  return n.toString(16).padStart(6, "0");
}
function zn(n, e) {
  const o = new Date(n);
  return `${o.toString().match(/\d\d:\d\d:\d\d/)[0]}${e === "ms" ? `.${String(o.getMilliseconds()).padStart(3, "0")}` : ""}`;
}
function Bn(n, e) {
  const o = /* @__PURE__ */ new Map(), t = Dt(n, e.toLowerCase(), o, 0);
  return o.clear(), t;
}
const At = 10;
function Dt(n, e, o, t) {
  if (t > At)
    return !1;
  let s = !1;
  const l = Object.keys(n);
  let i, u;
  for (let c = 0; c < l.length && (i = l[c], u = n[i], s = Pt(e, i, u, o, t + 1), !s); c++)
    ;
  return s;
}
function Pt(n, e, o, t, s) {
  let l = !1, i;
  return e === "_custom" && (e = o.display, o = o.value), (i = Zn(o)) && (o = i), e && It(e, n) ? (l = !0, t.set(o, !0)) : t.has(o) ? l = t.get(o) : Array.isArray(o) ? (t.set(o, null), l = Fn(o, n, t, s), t.set(o, l)) : on(o) ? (t.set(o, null), l = Dt(o, n, t, s), t.set(o, l)) : It(o, n) && (l = !0, t.set(o, !0)), l;
}
function It(n, e) {
  return `${n}`.toLowerCase().includes(e);
}
function Zn(n) {
  return n === null ? "null" : n === ln ? "undefined" : n === sn ? "NaN" : n === an ? "Infinity" : n === rn ? "-Infinity" : !1;
}
function Fn(n, e, o, t) {
  if (t > At)
    return !1;
  let s = !1, l;
  for (let i = 0; i < n.length && (l = n[i], s = Pt(e, null, l, o, t + 1), !s); i++)
    ;
  return s;
}
const Mt = /* @__PURE__ */ B({
  __name: "NodeTag",
  props: {
    tag: {}
  },
  setup(n) {
    return (e, o) => J((p(), m("span", {
      style: Ie({
        color: `#${d(kt)(e.tag.textColor)}`,
        backgroundColor: `#${d(kt)(e.tag.backgroundColor)}`
      }),
      class: "ml-2 rounded-sm px-1 text-[0.75rem] leading-snug"
    }, [
      ne(le(e.tag.label), 1)
    ], 4)), [
      [d(Q), {
        content: e.tag.tooltip,
        html: !0
      }]
    ]);
  }
}), Gn = { class: "p2" }, Kn = ["onClick"], Wn = /* @__PURE__ */ B({
  __name: "SelectiveList",
  props: /* @__PURE__ */ Be({
    data: {}
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ Be(["select"], ["update:modelValue"]),
  setup(n, { emit: e }) {
    const o = e, t = et(n, "modelValue");
    function s(l) {
      t.value = l, o("select", l);
    }
    return (l, i) => (p(), m("ul", Gn, [
      (p(!0), m(te, null, ae(l.data, (u) => (p(), m("li", {
        key: u.id,
        class: ie(["selectable-item", { active: u.id === t.value }]),
        onClick: (c) => s(u.id)
      }, [
        ne(le(u.label) + " ", 1),
        (p(!0), m(te, null, ae(u.tags, (c, C) => (p(), z(Mt, {
          key: C,
          tag: c
        }, null, 8, ["tag"]))), 128))
      ], 10, Kn))), 128))
    ]));
  }
}), at = /* @__PURE__ */ B({
  __name: "ToggleExpanded",
  props: {
    value: { type: Boolean }
  },
  setup(n) {
    return (e, o) => (p(), m("i", {
      class: ie(["i-radix-icons:triangle-right flex-none text-4 op-50 transition-base", {
        "transform rotate-90": e.value
      }])
    }, null, 2));
  }
}), Ht = Symbol("StateEditorSymbol");
function qn(n) {
  const e = T(n);
  return ge(Ht, e), {
    context: e
  };
}
function it() {
  return ve(Ht);
}
function Jn() {
  const n = T(""), e = T("string"), o = T(!1), t = it();
  return {
    editingText: n,
    editing: o,
    toggleEditing(s) {
      s && (e.value = s), o.value = !o.value;
    },
    editingType: e,
    nodeId: V(() => t.value.nodeId)
  };
}
function Yn(n, e) {
  if (n === "array")
    return e.length;
  const o = "newProp";
  let t = 1;
  for (; ; ) {
    const s = `${o}${t}`;
    if (!e[s])
      return s;
    t++;
  }
}
function Xn() {
  const n = T({
    enable: !1,
    key: "",
    value: "undefined"
  });
  function e(t, s) {
    const l = Yn(t, s);
    n.value = {
      enable: !0,
      key: l.toString(),
      value: "undefined"
    };
  }
  function o() {
    n.value = {
      enable: !1,
      key: "",
      value: "undefined"
    };
  }
  return {
    addNewProp: e,
    resetDrafting: o,
    draftingNewProp: n
  };
}
const Lt = "expanded-state";
function me(n = "") {
  const e = T([]);
  return ge(`${Lt}-${n}`, e), {
    expanded: e
  };
}
function rt(n = "") {
  const e = ve(`${Lt}-${n}`, T([]));
  function o(t) {
    const s = e.value.indexOf(t);
    s === -1 ? e.value.push(t) : e.value.splice(s, 1);
  }
  return {
    expanded: e,
    toggleExpanded: o
  };
}
function Qn(n, e = {}) {
  const {
    enter: o = () => {
    },
    leave: t = () => {
    },
    initial: s = !1
  } = e, l = T(s);
  return he(n, "mouseenter", () => {
    l.value = !0, o();
  }), he(n, "mouseleave", () => {
    l.value = !1, t();
  }), {
    isHovering: l
  };
}
const eo = { class: "flex pl5px" }, to = { class: "w160px py5px" }, no = /* @__PURE__ */ B({
  __name: "StateFieldEditor",
  props: {
    data: {},
    hovering: { type: Boolean },
    depth: {},
    showAddIfNeeded: { type: Boolean, default: !0 },
    disableEdit: { type: Boolean }
  },
  emits: ["enableEditInput", "addNewProp"],
  setup(n) {
    const e = n, o = it(), { copy: t, isSupported: s } = Pn(), l = T(!1), i = V(() => Et(e.data.value)), u = V(() => i.value.value), c = V(() => i.value.customType), C = V(() => u.value === null ? "null" : typeof u.value), v = {
      flat: !0,
      size: "mini"
    }, a = V(() => ({
      "opacity-0": !e.hovering
    }));
    async function f(b, h = !1) {
      await D.value.editInspectorState({
        path: e.data.path || [e.data.key],
        inspectorId: o.value.inspectorId,
        type: e.data.stateType,
        nodeId: o.value.nodeId,
        state: {
          newKey: null,
          value: Tt(b),
          type: C.value,
          remove: h
        }
      }), await D.value.sendInspectorState(o.value.inspectorId);
    }
    function g(b, h) {
      const S = typeof b == "number" ? b + h : BigInt(b) + BigInt(h);
      f(S);
    }
    return (b, h) => (p(), m("div", eo, [
      !e.disableEdit && b.data.editable ? (p(), m(te, { key: 0 }, [
        C.value === "string" || C.value === "number" || C.value === "object" || C.value === "null" ? (p(), m(te, { key: 0 }, [
          J((p(), z(d(de), ke(v, {
            class: a.value,
            onClick: h[0] || (h[0] = fe((S) => b.$emit("enableEditInput", C.value), ["stop"]))
          }), {
            icon: M(() => [
              x(d(oe), { icon: "i-material-symbols-edit-rounded" })
            ]),
            _: 1
          }, 16, ["class"])), [
            [d(Q), {
              content: "Edit value"
            }]
          ]),
          C.value === "object" && b.showAddIfNeeded ? J((p(), z(d(de), ke({ key: 0 }, v, {
            class: a.value,
            onClick: h[1] || (h[1] = fe((S) => b.$emit("addNewProp", Array.isArray(u.value) ? "array" : "object"), ["stop"]))
          }), {
            icon: M(() => [
              x(d(oe), { icon: "i-material-symbols-add-circle-rounded" })
            ]),
            _: 1
          }, 16, ["class"])), [
            [d(Q), {
              content: "Add new value"
            }]
          ]) : U("", !0)
        ], 64)) : U("", !0),
        C.value === "boolean" ? (p(), z(d(de), ke({ key: 1 }, v, {
          class: a.value,
          onClick: h[2] || (h[2] = (S) => f(!u.value))
        }), {
          icon: M(() => [
            x(d(oe), {
              icon: u.value ? "i-material-symbols-check-box-sharp" : "i-material-symbols-check-box-outline-blank-sharp"
            }, null, 8, ["icon"])
          ]),
          _: 1
        }, 16, ["class"])) : C.value === "number" || c.value === "bigint" ? (p(), m(te, { key: 2 }, [
          x(d(de), ke(v, {
            class: a.value,
            onClick: h[3] || (h[3] = fe((S) => g(u.value, 1), ["stop"]))
          }), {
            icon: M(() => [
              x(d(oe), { icon: "i-carbon-add" })
            ]),
            _: 1
          }, 16, ["class"]),
          x(d(de), ke(v, {
            class: a.value,
            onClick: h[4] || (h[4] = fe((S) => g(u.value, -1), ["stop"]))
          }), {
            icon: M(() => [
              x(d(oe), { icon: "i-carbon-subtract" })
            ]),
            _: 1
          }, 16, ["class"])
        ], 64)) : U("", !0)
      ], 64)) : U("", !0),
      !e.disableEdit && b.depth > 0 && b.data.editable ? (p(), z(d(de), ke({ key: 1 }, v, {
        class: a.value,
        onClick: h[5] || (h[5] = fe((S) => f(u.value, !0), ["stop"]))
      }), {
        icon: M(() => [
          x(d(oe), { icon: "i-material-symbols-delete-rounded" })
        ]),
        _: 1
      }, 16, ["class"])) : U("", !0),
      x(d(pn), {
        class: ie({
          "opacity-0": !b.hovering && !l.value
        }),
        "button-props": {
          flat: !0,
          size: "mini"
        },
        disabled: !d(s),
        "onUpdate:visible": h[8] || (h[8] = (S) => l.value = S)
      }, {
        popper: M(() => [
          r("div", to, [
            x(d(xt), {
              onClick: h[6] || (h[6] = (S) => d(t)(typeof u.value == "object" ? JSON.stringify(u.value) : u.value.toString()))
            }, {
              icon: M(() => [
                x(d(oe), {
                  icon: "i-material-symbols-copy-all-rounded",
                  class: "mt4px"
                }),
                h[9] || (h[9] = ne(" Copy Value "))
              ]),
              _: 1
            }),
            x(d(xt), {
              onClick: h[7] || (h[7] = () => {
                d(t)(b.data.key);
              })
            }, {
              icon: M(() => [
                x(d(oe), {
                  icon: "i-material-symbols-copy-all-rounded",
                  class: "mt4px"
                }),
                h[10] || (h[10] = ne(" Copy Path "))
              ]),
              _: 1
            })
          ])
        ]),
        "button-icon": M(() => [
          x(d(oe), { icon: "i-material-symbols:more-vert" })
        ]),
        _: 1
      }, 8, ["class", "disabled"])
    ]));
  }
}), oo = { class: "flex-inline items-center gap4px" }, Ye = /* @__PURE__ */ B({
  __name: "StateFieldInputEditor",
  props: {
    modelValue: {},
    customType: {},
    showActions: { type: Boolean, default: !0 },
    autoFocus: { type: Boolean, default: !0 }
  },
  emits: ["cancel", "submit", "update:modelValue"],
  setup(n, { emit: e }) {
    const o = n, t = e, s = V(() => o.customType === "date" ? "datetime-local" : ""), { escape: l, enter: i } = jn();
    tt(() => {
      l.value ? t("cancel") : i.value && t("submit");
    });
    const u = Un(o, "modelValue", t);
    function c(a) {
      try {
        return Ue(a, o.customType), !0;
      } catch {
        return !1;
      }
    }
    const C = T(!1);
    return X(u, _n(() => {
      C.value = !c(u.value);
    }, 300)), (a, f) => (p(), m("span", oo, [
      x(d(Ze), {
        modelValue: d(u),
        "onUpdate:modelValue": f[0] || (f[0] = (g) => Ge(u) ? u.value = g : null),
        type: s.value,
        variant: C.value ? "warning" : "normal",
        class: ie(["h25px px4px", a.customType === "date" ? "w240px" : "w120px"]),
        "auto-focus": a.autoFocus,
        onClick: f[1] || (f[1] = fe(() => {
        }, ["stop"]))
      }, null, 8, ["modelValue", "type", "variant", "class", "auto-focus"]),
      a.showActions ? (p(), m(te, { key: 0 }, [
        J((p(), z(d(de), {
          size: "mini",
          flat: "",
          class: "p2px!",
          onClick: f[2] || (f[2] = fe((g) => a.$emit("cancel"), ["stop"]))
        }, {
          icon: M(() => [
            x(d(oe), { icon: "i-material-symbols-cancel" })
          ]),
          _: 1
        })), [
          [d(Q), {
            content: "Esc to cancel"
          }]
        ]),
        C.value ? J((p(), z(d(oe), {
          key: 1,
          icon: "i-material-symbols-warning",
          class: "color-warning-500 dark:color-warning-300"
        }, null, 512)), [
          [d(Q), {
            content: "Invalid value"
          }]
        ]) : J((p(), z(d(de), {
          key: 0,
          size: "mini",
          flat: "",
          class: "p2px!",
          onClick: f[3] || (f[3] = fe((g) => a.$emit("submit"), ["stop"]))
        }, {
          icon: M(() => [
            x(d(oe), { icon: "i-material-symbols-save" })
          ]),
          _: 1
        })), [
          [d(Q), {
            content: "Enter to submit change"
          }]
        ])
      ], 64)) : U("", !0)
    ]));
  }
}), lo = {
  key: 1,
  pl5: ""
}, so = {
  op70: "",
  class: "whitespace-nowrap"
}, ao = ["innerHTML"], io = { key: 0 }, ro = {
  "overflow-hidden": "",
  "text-ellipsis": "",
  "whitespace-nowrap": "",
  "state-key": ""
}, $t = 30, uo = /* @__PURE__ */ B({
  __name: "StateFieldViewer",
  props: {
    data: {},
    depth: {},
    index: {},
    expandedStateId: {}
  },
  setup(n) {
    const e = n, o = T($t), t = V(() => un(e.data.value, !1, {
      customClass: {
        string: "max-w-120 truncate"
      }
    })), s = V(() => ht(e.data.value)), l = V(() => Et(e.data.value)), { expanded: i, toggleExpanded: u } = rt(e.expandedStateId ?? ""), c = V(() => {
      var _;
      return s.value === "custom" ? `${((_ = e.data.value._custom) == null ? void 0 : _.type) ?? "string"}-custom-state` : "";
    }), C = V(() => {
      const { value: _ } = l.value;
      return gt(_) ? _.length : Ct(_) ? Object.keys(_).length : 0;
    }), v = V(() => e.data.path || [e.data.key]), a = V(() => v.value[v.value.length - 1]), f = V(() => {
      var F, se, re, ce, Ce, _e, He;
      const _ = ["Reactive"], N = ((se = (F = e.data.value) == null ? void 0 : F._custom) == null ? void 0 : se.stateTypeName) || ((re = e.data) == null ? void 0 : re.stateTypeName);
      if (_.includes(N))
        return N;
      if ((Ce = (ce = e.data.value) == null ? void 0 : ce.fields) != null && Ce.abstract)
        return "";
      {
        const Le = (He = (_e = e.data.value) == null ? void 0 : _e._custom) == null ? void 0 : He.type, Oe = s.value === "custom" && !Le ? `"${t.value}"` : t.value === "" ? '""' : t.value, je = s.value === "custom" && Le === "ref" ? ht(Oe) : s.value, We = s.value === "string" ? "select-text" : "", Ee = `<span class="${je}-state-type flex whitespace-nowrap ${We}">${Oe}</span>`;
        return N ? `${Ee} <span class="text-gray-500">(${N})</span>` : Ee;
      }
    }), g = V(() => {
      const { value: _, inherit: N, customType: F } = l.value, se = F === "set";
      let re = [];
      return gt(_) ? _.slice(0, o.value).map((Ce, _e) => ({
        key: _e.toString(),
        path: [...v.value, _e.toString()],
        value: Ce,
        ...N,
        editable: e.data.editable && !se,
        creating: !1
      })) : (Ct(_) && (re = Object.keys(_).slice(0, o.value).map((ce) => ({
        key: ce,
        path: [...v.value, ce],
        value: _[ce],
        ...N,
        editable: e.data.editable && !se,
        creating: !1
      })), s.value !== "custom" && (re = Vt(re))), re === e.data.value ? [] : re);
    }), b = V(() => g.value.length > 0), h = T(), S = it(), { isHovering: j } = Qn(() => h.value), { editingType: O, editing: A, editingText: k, toggleEditing: R, nodeId: H } = Jn();
    X(() => A.value, (_) => {
      if (_) {
        const { value: N } = l.value;
        k.value = cn(N, l.value.customType);
      } else
        k.value = "";
    });
    async function K() {
      const _ = e.data;
      await D.value.editInspectorState({
        path: v.value,
        inspectorId: S.value.inspectorId,
        type: _.stateType,
        nodeId: H.value,
        state: {
          newKey: null,
          type: O.value,
          value: Ue(k.value, l.value.customType)
        }
      }), await D.value.sendInspectorState(S.value.inspectorId), R();
    }
    const { addNewProp: Z, draftingNewProp: E, resetDrafting: P } = Xn();
    function w(_) {
      const N = `${e.depth}-${e.index}`;
      i.value.includes(N) || u(N), Z(_, l.value.value);
    }
    async function I() {
      const _ = e.data;
      await D.value.editInspectorState({
        path: [...v.value, E.value.key],
        inspectorId: S.value.inspectorId,
        type: _.stateType,
        nodeId: H.value,
        state: {
          newKey: E.value.key,
          type: typeof Ue(E.value.value),
          value: Ue(E.value.value)
        }
      }), await D.value.sendInspectorState(S.value.inspectorId), P();
    }
    return (_, N) => (p(), m("div", null, [
      r("div", {
        ref_key: "containerRef",
        ref: h,
        class: ie(["font-state-field flex items-center text-3.5", [b.value && "cursor-pointer hover:bg-active"]]),
        style: Ie({ paddingLeft: `${_.depth * 15 + 4}px` }),
        onClick: N[1] || (N[1] = (F) => d(u)(`${_.depth}-${_.index}`))
      }, [
        b.value ? (p(), z(at, {
          key: 0,
          value: d(i).includes(`${_.depth}-${_.index}`)
        }, null, 8, ["value"])) : (p(), m("span", lo)),
        r("span", so, le(a.value), 1),
        N[5] || (N[5] = r("span", { mx1: "" }, ":", -1)),
        d(A) ? (p(), z(Ye, {
          key: 2,
          modelValue: d(k),
          "onUpdate:modelValue": N[0] || (N[0] = (F) => Ge(k) ? k.value = F : null),
          class: "mr-1",
          "custom-type": l.value.customType,
          onCancel: d(R),
          onSubmit: K
        }, null, 8, ["modelValue", "custom-type", "onCancel"])) : U("", !0),
        r("span", {
          class: ie([c.value, "flex whitespace-nowrap"])
        }, [
          r("span", {
            class: "flex",
            innerHTML: f.value
          }, null, 8, ao)
        ], 2),
        x(no, {
          hovering: d(j),
          "disable-edit": d(S).disableEdit || d(A),
          data: _.data,
          depth: _.depth,
          onEnableEditInput: d(R),
          onAddNewProp: w
        }, null, 8, ["hovering", "disable-edit", "data", "depth", "onEnableEditInput"])
      ], 6),
      b.value && d(i).includes(`${_.depth}-${_.index}`) ? (p(), m("div", io, [
        x(Ot, {
          data: g.value,
          depth: _.depth,
          index: _.index
        }, null, 8, ["data", "depth", "index"]),
        C.value > o.value ? J((p(), z(d(de), {
          key: 0,
          flat: "",
          size: "mini",
          class: "ml-4",
          onClick: N[2] || (N[2] = (F) => o.value += $t)
        }, {
          icon: M(() => [
            x(d(oe), { icon: "i-material-symbols-more-horiz" })
          ]),
          _: 1
        })), [
          [d(Q), "Show more"]
        ]) : U("", !0),
        d(E).enable ? (p(), m("div", {
          key: 1,
          style: Ie({ paddingLeft: `${(_.depth + 1) * 15 + 4}px` })
        }, [
          r("span", ro, [
            x(Ye, {
              modelValue: d(E).key,
              "onUpdate:modelValue": N[3] || (N[3] = (F) => d(E).key = F),
              "show-actions": !1
            }, null, 8, ["modelValue"])
          ]),
          N[6] || (N[6] = r("span", { "mx-1": "" }, ":", -1)),
          x(Ye, {
            modelValue: d(E).value,
            "onUpdate:modelValue": N[4] || (N[4] = (F) => d(E).value = F),
            "auto-focus": !1,
            onCancel: d(P),
            onSubmit: I
          }, null, 8, ["modelValue", "onCancel"])
        ], 4)) : U("", !0)
      ])) : U("", !0)
    ]));
  }
}), be = (n, e) => {
  const o = n.__vccOpts || n;
  for (const [t, s] of e)
    o[t] = s;
  return o;
}, co = /* @__PURE__ */ be(uo, [["__scopeId", "data-v-f3246652"]]), Ot = /* @__PURE__ */ B({
  __name: "ChildStateViewer",
  props: {
    data: {},
    depth: { default: 0 },
    index: {},
    expandedStateId: { default: "" }
  },
  setup(n) {
    return (e, o) => (p(), m("div", null, [
      (p(!0), m(te, null, ae(e.data, (t, s) => (p(), m("div", { key: s }, [
        x(co, {
          data: t,
          depth: e.depth + 1,
          index: `${e.index}-${s}`,
          "expanded-state-id": e.expandedStateId
        }, null, 8, ["data", "depth", "index", "expanded-state-id"])
      ]))), 128))
    ]));
  }
}), po = ["onClick"], fo = {
  key: 1,
  pl5: ""
}, vo = {
  "font-state-field": "",
  "text-3.5": ""
}, mo = { key: 0 }, De = /* @__PURE__ */ B({
  __name: "RootStateViewer",
  props: {
    data: {},
    nodeId: {},
    inspectorId: {},
    disableEdit: { type: Boolean, default: !1 },
    expandedStateId: { default: "" }
  },
  setup(n) {
    const e = n;
    function o() {
      return {
        nodeId: e.nodeId,
        inspectorId: e.inspectorId,
        disableEdit: e.disableEdit
      };
    }
    const { context: t } = qn(o());
    tt(() => {
      t.value = o();
    });
    const { expanded: s, toggleExpanded: l } = rt(e.expandedStateId);
    return (i, u) => (p(), m("div", null, [
      (p(!0), m(te, null, ae(i.data, (c, C, v) => (p(), m("div", { key: v }, [
        r("div", {
          class: ie(["flex items-center", [(c == null ? void 0 : c.length) && "cursor-pointer hover:bg-active"]]),
          onClick: (a) => d(l)(`${v}`)
        }, [
          c != null && c.length ? (p(), z(at, {
            key: 0,
            value: d(s).includes(`${v}`)
          }, null, 8, ["value"])) : (p(), m("span", fo)),
          r("span", vo, le(C), 1)
        ], 10, po),
        c != null && c.length && d(s).includes(`${v}`) ? (p(), m("div", mo, [
          x(Ot, {
            data: c,
            index: `${v}`,
            "expanded-state-id": i.expandedStateId
          }, null, 8, ["data", "index", "expanded-state-id"])
        ])) : U("", !0)
      ]))), 128))
    ]));
  }
}), jt = Symbol("SelectedSymbolKey");
function ho() {
  const n = T("");
  return ge(jt, n), {
    selected: n
  };
}
function go() {
  const n = ve(jt, T(""));
  function e(o) {
    n.value = o;
  }
  return {
    selected: n,
    select: e
  };
}
const Co = ["onClick", "onDblclick"], _o = {
  key: 1,
  pl5: ""
}, xo = {
  "font-state-field": "",
  "text-3.5": ""
}, yo = {
  key: 0,
  class: "text-gray-400 dark:text-gray-600 group-hover:text-white group-hover:op50 [.active_&]:op50 [.active_&]:text-white!"
}, bo = {
  "group-hover:text-white": "",
  class: "ws-nowrap [.active_&]:text-white"
}, wo = {
  key: 1,
  class: "text-gray-400 dark:text-gray-600 group-hover:text-white group-hover:op50 [.active_&]:op50 [.active_&]:text-white!"
}, ko = { key: 0 }, Pe = /* @__PURE__ */ B({
  __name: "TreeViewer",
  props: /* @__PURE__ */ Be({
    data: {},
    depth: { default: 0 },
    withTag: { type: Boolean, default: !1 }
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(n) {
    const e = et(n, "modelValue"), { expanded: o, toggleExpanded: t } = rt();
    go();
    function s(i) {
      return "name" in i && (i == null ? void 0 : i.name) || "label" in i && i.label;
    }
    function l(i) {
      e.value = i;
    }
    return (i, u) => (p(!0), m(te, null, ae(i.data, (c, C) => {
      var v, a;
      return p(), m("div", {
        key: C,
        class: ie({
          "min-w-max": i.depth === 0
        })
      }, [
        r("div", {
          class: ie(["group flex cursor-pointer items-center rounded-1 hover:bg-primary-300 hover:dark:bg-gray-600", { "bg-primary-600! active": e.value === c.id }]),
          style: Ie({ paddingLeft: `${15 * i.depth + 4}px` }),
          onClick: (f) => l(c.id),
          onDblclick: (f) => d(t)(c.id)
        }, [
          (v = c == null ? void 0 : c.children) != null && v.length ? (p(), z(at, {
            key: 0,
            value: d(o).includes(c.id),
            class: "[.active_&]:op20 group-hover:op20",
            onClick: fe((f) => d(t)(c.id), ["stop"])
          }, null, 8, ["value", "onClick"])) : (p(), m("span", _o)),
          r("span", xo, [
            i.withTag ? (p(), m("span", yo, "<")) : U("", !0),
            r("span", bo, le(s(c)), 1),
            i.withTag ? (p(), m("span", wo, ">")) : U("", !0)
          ]),
          (p(!0), m(te, null, ae(c.tags, (f, g) => (p(), z(Mt, {
            key: g,
            tag: f
          }, null, 8, ["tag"]))), 128))
        ], 46, Co),
        (a = c == null ? void 0 : c.children) != null && a.length && d(o).includes(c.id) ? (p(), m("div", ko, [
          x(Pe, {
            modelValue: e.value,
            "onUpdate:modelValue": u[0] || (u[0] = (f) => e.value = f),
            data: c == null ? void 0 : c.children,
            depth: i.depth + 1,
            "with-tag": i.withTag
          }, null, 8, ["modelValue", "data", "depth", "with-tag"])
        ])) : U("", !0)
      ], 2);
    }), 128));
  }
}), Io = { class: "absolute left-0 top-0 h-full w-full flex flex-col rounded-br-2.5 rounded-tr-2.5 bg-white p2 dark:bg-[#121212]" }, $o = { class: "flex-1 overflow-scroll text-3.5" }, So = /* @__PURE__ */ B({
  __name: "RenderCode",
  props: {
    code: {}
  },
  emits: ["close"],
  setup(n, { emit: e }) {
    const o = e;
    function t() {
      o("close");
    }
    return (s, l) => (p(), m("div", Io, [
      r("div", { class: "h-12 w-full flex items-center justify-between p-2" }, [
        l[0] || (l[0] = r("span", { class: "font-500" }, "Render Code", -1)),
        r("i", {
          class: "i-carbon-close cursor-pointer hover:op80",
          onClick: t
        })
      ]),
      r("div", $o, [
        x(d(fn), {
          code: s.code,
          lang: "javascript"
        }, null, 8, ["code"])
      ])
    ]));
  }
}), To = { class: "h-full w-full" }, Eo = { class: "no-scrollbar h-full flex select-none gap-2 overflow-scroll" }, Vo = {
  key: 0,
  class: "h-full flex flex-col p2"
}, No = { class: "flex py2" }, Ro = {
  xmlns: "http://www.w3.org/2000/svg",
  style: { height: "1.1em", width: "1.1em" },
  class: "op-80 hover:op-100",
  viewBox: "0 0 24 24"
}, Ao = { class: "h-full flex flex-col p2" }, Do = { class: "flex py2" }, Po = {
  key: 0,
  class: "font-state-field flex items-center px-1 text-3.5"
}, Mo = {
  "group-hover:text-white": "",
  class: "max-w-40 of-hidden text-ellipsis ws-nowrap [.active_&]:text-white"
}, Ho = { class: "flex items-center gap-2 px-1" }, Lo = { class: "h-full flex flex-col items-center justify-center gap-2" }, Oo = { class: "block" }, jo = {
  xmlns: "http://www.w3.org/2000/svg",
  style: { height: "2em", width: "2em", opacity: "0.5", color: "#00dc82" },
  class: "animate-fade",
  viewBox: "0 0 24 24"
}, Uo = { class: "flex items-center justify-center" }, Ve = "components", zo = /* @__PURE__ */ B({
  __name: "index",
  emits: ["openInEditor", "onInspectComponentStart", "onInspectComponentEnd"],
  setup(n, { emit: e }) {
    const o = e, t = T(), s = T(!1), { width: l } = Ln(t), i = V(() => s.value ? l.value < 700 : !1), u = T(""), c = T(""), [C, v] = Rn(!0), a = T(!1), f = T(!1), g = T(""), b = T(!1);
    function h($, y = [], L = []) {
      var G;
      return y.push($.id), ((G = $.children) == null ? void 0 : G.length) === 0 && L.push([...y]), Array.isArray($.children) && $.children.forEach((q) => {
        h(q, y, L);
      }), y.pop(), L;
    }
    function S($) {
      const y = [], L = (G) => {
        G == null || G.forEach((q) => {
          var we;
          y.push(q), (we = q.children) != null && we.length && L(q.children);
        });
      };
      return L($), y;
    }
    function j($, y) {
      const L = [];
      return $ == null || $.forEach((G) => {
        L.push(...G.slice(0, y + 1));
      }), [...new Set(L)];
    }
    function O($, y) {
      const L = [];
      return $.forEach((G) => {
        const q = G.indexOf(y);
        q !== -1 && L.push(...G.slice(0, q + 1));
      }), [...new Set(L)];
    }
    const A = T([]), k = V(() => {
      var $, y;
      return ($ = A.value) != null && $.length ? h((y = A.value) == null ? void 0 : y[0]) : [];
    }), R = V(() => S(A.value)), H = V(() => R.value.map(($) => $.id)), K = T({}), Z = T(""), E = V(() => {
      const $ = [], y = (L) => {
        L.forEach((G) => {
          var q;
          G.id === Z.value && $.push(G), (q = G.children) != null && q.length && y(G.children);
        });
      };
      return y(A.value), $[0];
    }), P = V(() => {
      var $;
      return (($ = E.value) == null ? void 0 : $.file) ?? "";
    }), w = V(() => {
      const $ = {};
      for (const y in K.value) {
        const G = K.value[y].filter((we) => {
          try {
            return Bn({
              [we.key]: we.value
            }, c.value);
          } catch (qt) {
            return {
              [we.key]: qt
            };
          }
        }), q = gn(Object.values(Cn(Vt(G), "stateType")));
        G.length && ($[y] = q);
      }
      return $;
    }), { expanded: I } = me(), { expanded: _ } = me("component-state");
    ho();
    async function N($ = "") {
      return D.value.getInspectorTree({ inspectorId: Ve, filter: $ }).then((y) => {
        var G, q;
        const L = Y(y);
        A.value = L, Z.value = (q = (G = A.value) == null ? void 0 : G[0]) == null ? void 0 : q.id, I.value = j(k.value, 1), a.value = !0;
      });
    }
    function F($) {
      if (!$ || !($ != null && $.state))
        return {};
      const y = {};
      return $.state.forEach((L) => {
        y[L.type] || (y[L.type] = []), y[L.type].push(L);
      }), y;
    }
    function se($) {
      D.value.getInspectorState({ inspectorId: Ve, nodeId: $ }).then((y) => {
        const L = Y(y);
        L && (K.value = F(L), _.value = Array.from({ length: Object.keys(K.value).length }, (G, q) => `${q}`));
      });
    }
    X(Z, ($) => {
      se($), b.value && je();
    });
    function re($) {
      const y = Y($);
      y.inspectorId !== Ve || y.nodeId !== Z.value || (K.value = F({ state: y.state.state }));
    }
    D.functions.on(W.INSPECTOR_STATE_UPDATED, re), N();
    function ce($) {
      const y = $.trim().toLowerCase();
      v(), N(y).then(() => {
        v();
      });
    }
    An(u, ($) => {
      ce($);
    }, { debounce: 300 });
    function Ce($) {
      var L, G;
      const y = Y($);
      y.inspectorId === Ve && (u.value ? ce(u.value) : A.value = y.rootNodes, H.value.includes(Z.value) || (Z.value = (G = (L = A.value) == null ? void 0 : L[0]) == null ? void 0 : G.id, I.value = j(k.value, 1)));
    }
    D.functions.on(W.INSPECTOR_TREE_UPDATED, Ce), ye(() => {
      D.functions.off(W.INSPECTOR_STATE_UPDATED, re), D.functions.off(W.INSPECTOR_TREE_UPDATED, Ce);
    });
    function _e() {
      f.value = !0, o("onInspectComponentStart"), D.value.inspectComponentInspector().then(($) => {
        const y = JSON.parse($);
        Z.value = y.id, I.value.includes(y.id) || I.value.push(y.id), I.value = [.../* @__PURE__ */ new Set([...I.value, ...O(k.value, y.id)])], Ft();
      }).finally(() => {
        f.value = !1, o("onInspectComponentEnd");
      });
    }
    function He() {
      f.value = !1, D.value.cancelInspectComponentInspector();
    }
    function Le() {
      D.value.scrollToComponent(Z.value);
    }
    function Oe() {
      D.value.inspectDOM(Z.value).then(() => {
        chrome.devtools.inspectedWindow.eval("inspect(window.__VUE_DEVTOOLS_INSPECT_DOM_TARGET__)");
      });
    }
    function je() {
      D.value.getComponentRenderCode(Z.value).then(($) => {
        g.value = $, b.value = !0;
      });
    }
    function We() {
      o("openInEditor", P.value);
    }
    const Ee = T();
    function Ft() {
      setTimeout(() => {
        var y;
        const $ = (y = Ee.value) == null ? void 0 : y.querySelector(".active");
        $ == null || $.scrollIntoView({ behavior: "smooth", block: "center" });
      }, 300);
    }
    function Gt() {
      g.value = "", b.value = !1;
    }
    const qe = Jt(), pt = V(() => qe.appRecords.value.map(($) => ({
      label: $.name + ($.version ? ` (${$.version})` : ""),
      value: $.id
    }))), Kt = V(() => pt.value.map(($) => ({
      label: $.label,
      id: $.value
    }))), Je = T(qe.activeAppRecordId.value);
    tt(() => {
      Je.value = qe.activeAppRecordId.value;
    });
    function Wt($) {
      D.value.toggleApp($).then(() => {
        Z.value = "", N();
      });
    }
    return ($, y) => (p(), m("div", To, [
      x(d(Ae), {
        ref_key: "splitpanesRef",
        ref: t,
        class: "flex-1 overflow-auto",
        horizontal: i.value,
        onReady: y[4] || (y[4] = (L) => s.value = !0)
      }, {
        default: M(() => [
          pt.value.length > 1 ? (p(), z(d(ue), {
            key: 0,
            border: "base h-full",
            size: "20"
          }, {
            default: M(() => [
              r("div", Eo, [
                x(Wn, {
                  modelValue: Je.value,
                  "onUpdate:modelValue": y[0] || (y[0] = (L) => Je.value = L),
                  data: Kt.value,
                  class: "w-full",
                  onSelect: Wt
                }, null, 8, ["modelValue", "data"])
              ])
            ]),
            _: 1
          })) : U("", !0),
          x(d(ue), {
            border: "base",
            "h-full": ""
          }, {
            default: M(() => [
              a.value ? (p(), m("div", Vo, [
                r("div", No, [
                  x(d(Ze), {
                    modelValue: u.value,
                    "onUpdate:modelValue": y[1] || (y[1] = (L) => u.value = L),
                    "loading-debounce-time": 250,
                    loading: !d(C),
                    placeholder: "Find components...",
                    class: "flex-1 text-3.5"
                  }, null, 8, ["modelValue", "loading"]),
                  d(dn) ? U("", !0) : J((p(), m("button", {
                    key: 0,
                    "px-1": "",
                    class: "hover:color-#00dc82",
                    onClick: _e
                  }, [
                    (p(), m("svg", Ro, y[6] || (y[6] = [
                      r("path", {
                        fill: "currentColor",
                        d: "M10.611 10.611a1 1 0 0 1 1.11-.208l8.839 3.889a1 1 0 0 1-.14 1.88l-3.338.91l-.91 3.338a1 1 0 0 1-1.88.14l-3.89-8.84a1 1 0 0 1 .209-1.109M17 3a3 3 0 0 1 3 3v3a1 1 0 1 1-2 0V6a1 1 0 0 0-1-1H6a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h3a1 1 0 1 1 0 2H6a3 3 0 0 1-3-3V6a3 3 0 0 1 3-3zm-3.73 10.269l1.715 3.9l.318-1.164a1 1 0 0 1 .701-.702l1.165-.318l-3.9-1.716Z"
                      }, null, -1)
                    ])))
                  ])), [
                    [
                      d(Q),
                      "Select component in the page",
                      void 0,
                      { bottom: !0 }
                    ]
                  ])
                ]),
                r("div", {
                  ref_key: "componentTreeContainer",
                  ref: Ee,
                  class: "no-scrollbar flex-1 select-none overflow-scroll"
                }, [
                  x(Pe, {
                    modelValue: Z.value,
                    "onUpdate:modelValue": y[2] || (y[2] = (L) => Z.value = L),
                    data: A.value,
                    "with-tag": !0
                  }, null, 8, ["modelValue", "data"])
                ], 512)
              ])) : U("", !0)
            ]),
            _: 1
          }),
          x(d(ue), {
            relative: "",
            "h-full": ""
          }, {
            default: M(() => {
              var L;
              return [
                r("div", Ao, [
                  r("div", Do, [
                    (L = E.value) != null && L.name ? (p(), m("span", Po, [
                      y[7] || (y[7] = r("span", { class: "text-gray-400 dark:text-gray-600" }, "<", -1)),
                      r("span", Mo, le(E.value.name), 1),
                      y[8] || (y[8] = r("span", { class: "text-gray-400 dark:text-gray-600" }, ">", -1))
                    ])) : U("", !0),
                    x(d(Ze), {
                      modelValue: c.value,
                      "onUpdate:modelValue": y[3] || (y[3] = (G) => c.value = G),
                      "loading-debounce-time": 250,
                      placeholder: "Filter State...",
                      class: "flex-1 text-3.5"
                    }, null, 8, ["modelValue"]),
                    r("div", Ho, [
                      J(r("i", {
                        class: "i-material-symbols-light:eye-tracking-outline h-4 w-4 cursor-pointer hover:op-70",
                        onClick: Le
                      }, null, 512), [
                        [
                          d(Q),
                          "Scroll to component",
                          void 0,
                          { bottom: !0 }
                        ]
                      ]),
                      J(r("i", {
                        class: "i-material-symbols-light:code h-5 w-5 cursor-pointer hover:op-70",
                        onClick: je
                      }, null, 512), [
                        [
                          d(Q),
                          "Show render code",
                          void 0,
                          { bottom: !0 }
                        ]
                      ]),
                      d(_t) ? J((p(), m("i", {
                        key: 0,
                        class: "i-material-symbols-light:menu-open h-5 w-5 cursor-pointer hover:op-70",
                        onClick: Oe
                      }, null, 512)), [
                        [
                          d(Q),
                          "Inspect DOM",
                          void 0,
                          { bottom: !0 }
                        ]
                      ]) : U("", !0),
                      P.value ? J((p(), m("i", {
                        key: 1,
                        class: "i-carbon-launch h-4 w-4 cursor-pointer hover:op-70",
                        onClick: We
                      }, null, 512)), [
                        [
                          d(Q),
                          "Open in Editor",
                          void 0,
                          { bottom: !0 }
                        ]
                      ]) : U("", !0)
                    ])
                  ]),
                  x(De, {
                    class: "no-scrollbar flex-1 select-none overflow-scroll",
                    data: w.value,
                    "node-id": Z.value,
                    "inspector-id": Ve,
                    "expanded-state-id": "component-state"
                  }, null, 8, ["data", "node-id"])
                ]),
                b.value && g.value ? (p(), z(So, {
                  key: 0,
                  code: g.value,
                  onClose: Gt
                }, null, 8, ["code"])) : U("", !0)
              ];
            }),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["horizontal"]),
      d(_t) ? (p(), z(d(vn), {
        key: 0,
        modelValue: f.value,
        "onUpdate:modelValue": y[5] || (y[5] = (L) => f.value = L),
        title: "",
        height: "12rem",
        closable: !1
      }, {
        footer: M(() => [
          r("div", Uo, [
            x(d(de), { onClick: He }, {
              default: M(() => y[11] || (y[11] = [
                ne(" Cancel ")
              ])),
              _: 1
            })
          ])
        ]),
        default: M(() => [
          r("div", Lo, [
            r("span", Oo, [
              (p(), m("svg", jo, y[9] || (y[9] = [
                r("g", {
                  fill: "none",
                  stroke: "currentColor",
                  "stroke-linecap": "round",
                  "stroke-linejoin": "round",
                  "stroke-width": "2"
                }, [
                  r("circle", {
                    cx: "12",
                    cy: "12",
                    r: ".5",
                    fill: "currentColor"
                  }),
                  r("path", { d: "M5 12a7 7 0 1 0 14 0a7 7 0 1 0-14 0m7-9v2m-9 7h2m7 7v2m7-9h2" })
                ], -1)
              ])))
            ]),
            y[10] || (y[10] = r("p", null, " Click on a component on the page to select it ", -1))
          ])
        ]),
        _: 1
      }, 8, ["modelValue"])) : U("", !0)
    ]));
  }
}), ql = /* @__PURE__ */ be(zo, [["__scopeId", "data-v-38589395"]]), Bo = {}, Zo = { class: "h-screen w-screen $ui-fcc flex-col" };
function Fo(n, e) {
  return p(), m("div", Zo, [
    e[0] || (e[0] = r("div", { class: "outer" }, [
      r("div", { class: "inner" }, [
        r("i", {
          class: "i-logos-vue inline-block h8! w8!",
          alt: "Vue logo"
        })
      ])
    ], -1)),
    nt(n.$slots, "default", {}, void 0, !0)
  ]);
}
const Go = /* @__PURE__ */ be(Bo, [["render", Fo], ["__scopeId", "data-v-b8018f07"]]), Ut = Symbol.for("VueDevToolsCustomInspectorStateSymbol");
function $e() {
  return ve(Ut);
}
function zt() {
  const n = T({
    homepage: "",
    id: "",
    label: "",
    logo: "",
    timelineLayerIds: []
  });
  return ge(Ut, n), n;
}
const ut = Symbol("VirtualRouteKey"), Bt = Symbol("VirtualRoutesKey");
function ct(n, e) {
  const o = (e == null ? void 0 : e.defaultRoutePath) ?? mt(n)[0].path, t = T({
    path: o
  }), s = V(() => mt(n)), l = V(() => t.value.path), i = B({
    setup() {
      return () => {
        const c = s.value.find((C) => C.path === l.value);
        return c ? en(c.component) : null;
      };
    }
  });
  function u() {
    t.value.path = o;
  }
  return ge(ut, t), ge(Bt, s), { VirtualRouterView: i, restoreRouter: u };
}
function Me() {
  const n = ve(ut);
  return {
    push(e) {
      n.value.path = e;
    }
  };
}
function Ko() {
  const n = ve(Bt), e = ve(ut);
  return {
    routes: n,
    currentRoute: e
  };
}
const Wo = {}, qo = {
  width: "175",
  height: "34",
  viewBox: "0 0 175 34",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};
function Jo(n, e) {
  return p(), m("svg", qo, e[0] || (e[0] = [
    r("path", {
      fill: "currentColor",
      d: "M0.570999 33.3121V1.80715H10.8288C13.6975 1.80715 16.3198 2.39225 18.6962 3.56242C21.0721 4.73263 22.9701 6.5029 24.3901 8.87327C25.81 11.2136 26.52 14.1091 26.52 17.5596C26.52 20.9801 25.81 23.8756 24.3901 26.2458C22.9701 28.6162 21.0721 30.3868 18.6962 31.5568C16.3198 32.7269 13.6975 33.3121 10.8288 33.3121H0.570999ZM10.1332 27.461C12.9439 27.461 15.2767 26.6812 17.1314 25.1208C18.9858 23.5306 19.913 21.0102 19.913 17.5596C19.913 14.1091 18.9858 11.5887 17.1314 9.99846C15.2767 8.40819 12.9439 7.61306 10.1332 7.61306H7.09085V27.461H10.1332ZM50.3628 25.6159C49.928 28.0463 48.7544 29.9966 46.8421 31.4669C44.9297 32.9368 42.6257 33.672 39.931 33.672C37.5551 33.672 35.5119 33.117 33.8024 32.007C32.1216 30.8668 30.8468 29.3965 29.9772 27.5961C29.1081 25.7961 28.6733 23.9056 28.6733 21.9253C28.6733 19.915 29.0645 18.0247 29.8468 16.2544C30.6584 14.4841 31.8756 13.0439 33.498 11.9337C35.1499 10.8236 37.1491 10.2685 39.4962 10.2685C41.9306 10.2685 43.9589 10.8086 45.5817 11.8888C47.2045 12.9689 48.3924 14.3491 49.1456 16.0294C49.928 17.7097 50.3192 19.4799 50.3192 21.3402C50.3192 22.0303 50.2904 22.7054 50.2324 23.3656H34.4979C34.7295 24.9858 35.3091 26.2612 36.2363 27.191C37.1927 28.0914 38.4243 28.5412 39.931 28.5412C41.1482 28.5412 42.1478 28.3014 42.9301 27.8214C43.7125 27.3111 44.2197 26.5759 44.4513 25.6159H50.3628ZM39.4962 14.8592C38.1055 14.8592 37.0043 15.2343 36.1931 15.9844C35.3815 16.7045 34.8455 17.8447 34.5847 19.4049H44.2777C44.1905 18.0847 43.7269 17.0045 42.8865 16.1644C42.0466 15.2943 40.9162 14.8592 39.4962 14.8592ZM64.1603 33.3121H58.5969L50.599 10.6285H56.7713L61.3788 25.3008L65.9427 10.6285H72.1582L64.1603 33.3121ZM88.2757 33.3121H81.7559V7.61306H72.7585V1.80715H97.2731V7.61306H88.2757V33.3121ZM106.325 33.672C104.123 33.672 102.138 33.1771 100.37 32.1868C98.6029 31.1667 97.2266 29.7717 96.2414 28.0011C95.2562 26.2011 94.7634 24.1907 94.7634 21.9703C94.7634 19.75 95.2562 17.7547 96.2414 15.9844C97.2266 14.1841 98.6029 12.7889 100.37 11.7987C102.138 10.7786 104.123 10.2685 106.325 10.2685C108.528 10.2685 110.513 10.7786 112.28 11.7987C114.048 12.7889 115.424 14.1841 116.409 15.9844C117.395 17.7547 117.887 19.75 117.887 21.9703C117.887 24.1907 117.395 26.2011 116.409 28.0011C115.424 29.7717 114.048 31.1667 112.28 32.1868C110.513 33.1771 108.528 33.672 106.325 33.672ZM106.325 28.2712C107.948 28.2712 109.266 27.6864 110.281 26.5159C111.295 25.3458 111.802 23.8306 111.802 21.9703C111.802 20.08 111.295 18.5498 110.281 17.3796C109.266 16.2094 107.948 15.6243 106.325 15.6243C104.702 15.6243 103.384 16.2094 102.37 17.3796C101.356 18.5498 100.848 20.08 100.848 21.9703C100.848 23.8306 101.356 25.3458 102.37 26.5159C103.384 27.6864 104.702 28.2712 106.325 28.2712ZM131.624 33.672C129.422 33.672 127.436 33.1771 125.669 32.1868C123.901 31.1667 122.525 29.7717 121.54 28.0011C120.555 26.2011 120.062 24.1907 120.062 21.9703C120.062 19.75 120.555 17.7547 121.54 15.9844C122.525 14.1841 123.901 12.7889 125.669 11.7987C127.436 10.7786 129.422 10.2685 131.624 10.2685C133.826 10.2685 135.811 10.7786 137.579 11.7987C139.346 12.7889 140.722 14.1841 141.708 15.9844C142.693 17.7547 143.186 19.75 143.186 21.9703C143.186 24.1907 142.693 26.2011 141.708 28.0011C140.722 29.7717 139.346 31.1667 137.579 32.1868C135.811 33.1771 133.826 33.672 131.624 33.672ZM131.624 28.2712C133.246 28.2712 134.565 27.6864 135.579 26.5159C136.593 25.3458 137.101 23.8306 137.101 21.9703C137.101 20.08 136.593 18.5498 135.579 17.3796C134.565 16.2094 133.246 15.6243 131.624 15.6243C130.001 15.6243 128.682 16.2094 127.668 17.3796C126.654 18.5498 126.147 20.08 126.147 21.9703C126.147 23.8306 126.654 25.3458 127.668 26.5159C128.682 27.6864 130.001 28.2712 131.624 28.2712ZM152.88 33.3121H146.838V0.771999H152.88V33.3121ZM165.652 33.672C162.957 33.672 160.769 32.9522 159.088 31.5117C157.437 30.0418 156.553 28.0914 156.437 25.661H161.653C161.768 26.6812 162.174 27.4913 162.87 28.0914C163.594 28.6613 164.522 28.9463 165.652 28.9463C166.579 28.9463 167.332 28.7363 167.912 28.3163C168.52 27.8963 168.825 27.3712 168.825 26.7412C168.825 25.9008 168.477 25.3159 167.782 24.9858C167.086 24.6557 165.985 24.3557 164.478 24.0856C162.913 23.7856 161.638 23.4556 160.653 23.0955C159.668 22.7354 158.813 22.0903 158.089 21.1602C157.393 20.2 157.045 18.8498 157.045 17.1096C157.045 15.7893 157.379 14.6192 158.045 13.599C158.741 12.5488 159.682 11.7387 160.871 11.1686C162.058 10.5685 163.391 10.2685 164.869 10.2685C167.506 10.2685 169.636 10.9436 171.259 12.2938C172.91 13.644 173.794 15.4293 173.91 17.6496H168.651C168.535 16.7195 168.115 15.9994 167.39 15.4893C166.695 14.9492 165.912 14.6792 165.043 14.6792C164.174 14.6792 163.478 14.8742 162.957 15.2643C162.435 15.6543 162.174 16.1944 162.174 16.8845C162.174 17.7246 162.508 18.2948 163.174 18.5948C163.87 18.8648 164.956 19.1049 166.434 19.3149C168.028 19.5549 169.331 19.855 170.346 20.215C171.389 20.5451 172.287 21.2052 173.041 22.1953C173.794 23.1855 174.171 24.6257 174.171 26.5159C174.171 28.6762 173.389 30.4166 171.824 31.737C170.288 33.0271 168.23 33.672 165.652 33.672Z"
    }, null, -1)
  ]));
}
const Yo = /* @__PURE__ */ be(Wo, [["render", Jo]]), Xo = {
  "h-full": "",
  "w-full": "",
  flex: "",
  "items-center": ""
}, Qo = {
  flex: "~ col gap2",
  ma: "",
  "px-5": ""
}, e1 = {
  flex: "~ col",
  "mt-20": "",
  "items-center": ""
}, t1 = {
  flex: "~",
  "mt--10": "",
  "items-center": "",
  "justify-center": ""
}, n1 = { class: "mr-1" }, o1 = {
  key: 0,
  class: "inline-block w20"
}, l1 = ["src", "alt"], s1 = {
  key: 1,
  class: "mr2 inline-block min-w-20 text-center text-10 font-600"
}, a1 = {
  mb6: "",
  "mt--1": "",
  "text-center": "",
  "text-sm": "",
  flex: "~ gap-1"
}, i1 = { op40: "" }, r1 = { flex: "~ gap2 wrap" }, u1 = {
  flex: "~ gap-6 wrap",
  "mt-5": "",
  "items-center": "",
  "justify-center": ""
}, c1 = ["href"], d1 = /* @__PURE__ */ B({
  __name: "About",
  setup(n) {
    const e = Me(), o = $e(), t = T(!1), s = T(!1);
    function l() {
      t.value = !0;
    }
    function i() {
      t.value = !0, s.value = !0;
    }
    return (u, c) => {
      var C;
      return p(), m("div", Xo, [
        r("div", Qo, [
          c[5] || (c[5] = r("div", { "flex-auto": "" }, null, -1)),
          r("div", e1, [
            r("div", t1, [
              r("span", n1, [
                t.value ? U("", !0) : (p(), m("span", o1)),
                J(r("img", {
                  src: d(o).logo,
                  alt: `${d(o).label} Logo`,
                  class: "inline-block h16",
                  onError: i,
                  onLoad: l
                }, null, 40, l1), [
                  [tn, t.value && !s.value]
                ]),
                s.value ? (p(), m("span", s1, le(d(o).label), 1)) : U("", !0)
              ]),
              r("span", null, [
                x(Yo, { "h-18": "" })
              ])
            ]),
            r("div", a1, [
              r("span", i1, le(d(o).label) + " DevTools ", 1)
            ])
          ]),
          c[6] || (c[6] = r("div", { "flex-auto": "" }, null, -1)),
          r("div", r1, [
            r("div", {
              flex: "~ col auto",
              "min-w-40": "",
              p4: "",
              "theme-card-lime": "",
              onClick: c[0] || (c[0] = (v) => d(e).push("/state"))
            }, c[2] || (c[2] = [
              r("div", {
                "i-carbon-tree-view-alt": "",
                "text-3xl": ""
              }, null, -1),
              r("code", null, "State", -1)
            ])),
            (C = d(o).timelineLayerIds) != null && C.length ? (p(), m("div", {
              key: 0,
              flex: "~ col auto",
              "min-w-40": "",
              p4: "",
              "theme-card-lime": "",
              onClick: c[1] || (c[1] = (v) => d(e).push("/timeline"))
            }, c[3] || (c[3] = [
              r("div", {
                "i-mdi:timeline-clock-outline": "",
                "text-3xl": ""
              }, null, -1),
              r("div", null, "Timeline", -1)
            ]))) : U("", !0)
          ]),
          r("div", u1, [
            r("a", {
              href: d(o).homepage,
              target: "_blank",
              flex: "~ gap1",
              "items-center": "",
              op50: "",
              hover: "op100 text-yellow",
              transition: ""
            }, c[4] || (c[4] = [
              r("div", { "i-carbon-document": "" }, null, -1),
              ne(" View Documentation ")
            ]), 8, c1)
          ]),
          c[7] || (c[7] = r("div", { "flex-auto": "" }, null, -1))
        ])
      ]);
    };
  }
}), p1 = {
  border: "b base",
  class: "flex items-center justify-between px3"
}, f1 = ["href"], v1 = ["href"], Se = /* @__PURE__ */ B({
  __name: "DevToolsHeader",
  props: {
    githubRepoLink: {},
    docLink: {}
  },
  setup(n) {
    const e = Me();
    return (o, t) => (p(), m("div", p1, [
      r("div", null, [
        nt(o.$slots, "default", {}, () => [
          r("i", {
            class: "i-ep:back cursor-pointer op70 text-base hover:op100",
            onClick: t[0] || (t[0] = (s) => d(e).push("/"))
          })
        ])
      ]),
      r("div", null, [
        r("a", {
          class: "pr2",
          href: o.docLink,
          target: "_blank",
          title: "View Documentation"
        }, t[1] || (t[1] = [
          r("i", { class: "i-clarity:document-line cursor-pointer op70 text-base hover:op100" }, null, -1)
        ]), 8, f1),
        o.githubRepoLink ? (p(), m("a", {
          key: 0,
          href: o.githubRepoLink,
          target: "_blank",
          title: "Star on GitHub"
        }, t[2] || (t[2] = [
          r("i", { class: "i-mdi:github cursor-pointer op70 text-base hover:op100" }, null, -1)
        ]), 8, v1)) : U("", !0)
      ])
    ]));
  }
}), m1 = { class: "flex gap3" }, h1 = ["onClick"], Te = /* @__PURE__ */ B({
  __name: "Navbar",
  setup(n) {
    const { routes: e, currentRoute: o } = Ko(), t = Me();
    return (s, l) => (p(), m("ul", m1, [
      (p(!0), m(te, null, ae(d(e), (i, u) => (p(), m("li", {
        key: u,
        class: ie(["h-10 flex cursor-pointer items-center justify-center border-b-[2px] border-solid text-size-sm leading-none hover:op100", d(o).path === i.path ? "op-100 font-medium text-primary-400 border-primary-400" : "border-transparent op-70"]),
        onClick: (c) => d(t).push(i.path)
      }, le(i.name), 11, h1))), 128))
    ]));
  }
}), g1 = { class: "flex-1 overflow-y-auto p2" }, C1 = { class: "max-w-[190px] flex-1 select-none py-1.5 text-sm" }, _1 = { class: "w-4/5" }, x1 = {
  key: 0,
  class: "flex justify-start"
}, y1 = { key: 1 }, Zt = /* @__PURE__ */ B({
  __name: "Settings",
  props: {
    pluginId: {},
    options: {},
    values: {}
  },
  emits: ["update"],
  setup(n, { emit: e }) {
    const o = n, t = e, s = V(() => o.options), l = V(() => o.values);
    function i(u, c) {
      D.value.updatePluginSettings(o.pluginId, u, c), D.value.getPluginSettings(o.pluginId).then((C) => {
        t("update", C);
      });
    }
    return (u, c) => (p(), m("div", g1, [
      r("ul", null, [
        (p(!0), m(te, null, ae(s.value, (C, v) => (p(), m("li", {
          key: v,
          class: "flex items-center py-2"
        }, [
          r("div", C1, le(C.label), 1),
          r("div", _1, [
            C.type === "boolean" ? (p(), m("div", x1, [
              x(d(mn), {
                "model-value": l.value[v],
                class: "row-reverse flex hover:bg-active py1 pl2 pr1",
                "onUpdate:modelValue": (a) => i(v, a)
              }, null, 8, ["model-value", "onUpdate:modelValue"])
            ])) : C.type === "choice" ? (p(), m("div", y1, [
              x(d(hn), {
                "model-value": l.value[v],
                options: C.options,
                "onUpdate:modelValue": (a) => i(v, a)
              }, null, 8, ["model-value", "options", "onUpdate:modelValue"])
            ])) : C.type === "text" ? (p(), z(d(Ze), {
              key: 2,
              "model-value": l.value[v],
              "onUpdate:modelValue": (a) => i(v, a)
            }, null, 8, ["model-value", "onUpdate:modelValue"])) : U("", !0)
          ])
        ]))), 128))
      ])
    ]));
  }
}), b1 = { class: "h-full flex flex-col" }, w1 = /* @__PURE__ */ B({
  __name: "Settings",
  setup(n) {
    const e = ve("pluginSettings"), o = $e(), t = V(() => e.value.options), s = V(() => e.value.values);
    function l(i) {
      e.value = i;
    }
    return (i, u) => (p(), m("div", b1, [
      x(Se, {
        "doc-link": d(o).homepage
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }, 8, ["doc-link"]),
      x(Zt, {
        "plugin-id": d(o).pluginId,
        options: t.value,
        values: s.value,
        onUpdate: l
      }, null, 8, ["plugin-id", "options", "values"])
    ]));
  }
}), k1 = {}, I1 = { class: "h-full flex flex-col items-center justify-center op50" };
function $1(n, e) {
  return p(), m("div", I1, [
    e[0] || (e[0] = r("i", { class: "i-lets-icons:blank-light" }, null, -1)),
    r("span", null, [
      nt(n.$slots, "default")
    ])
  ]);
}
const Ne = /* @__PURE__ */ be(k1, [["render", $1]]), S1 = { class: "h-full flex flex-col" }, T1 = { class: "h-full flex flex-col p2" }, E1 = {
  key: 0,
  class: "mb-1 flex justify-end pb-1",
  border: "b dashed base"
}, V1 = { class: "flex items-center gap-2 px-1" }, N1 = ["onClick"], R1 = { class: "no-scrollbar flex-1 select-none overflow-scroll" }, A1 = { class: "h-full flex flex-col p2" }, D1 = {
  key: 0,
  class: "mb-1 flex justify-end pb-1",
  border: "b dashed base"
}, P1 = { class: "flex items-center gap-2 px-1" }, M1 = ["onClick"], H1 = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const { expanded: e } = me(), { expanded: o } = me("custom-inspector-state"), t = $e(), s = V(() => t.value.id), l = T([]), i = T([]), u = T([]), c = V(() => {
      var w, I;
      return (w = u.value) != null && w.length ? b((I = u.value) == null ? void 0 : I[0]) : [];
    }), C = V(() => S(u.value)), v = V(() => C.value.map((w) => w.id)), a = T(""), f = T({}), g = V(() => !Object.keys(f.value).length);
    function b(w, I = [], _ = []) {
      var N;
      return I.push(w.id), ((N = w.children) == null ? void 0 : N.length) === 0 && _.push([...I]), Array.isArray(w.children) && w.children.forEach((F) => {
        b(F, I, _);
      }), I.pop(), _;
    }
    function h(w, I) {
      const _ = [];
      return w == null || w.forEach((N) => {
        _.push(...N.slice(0, I + 1));
      }), [...new Set(_)];
    }
    function S(w) {
      const I = [], _ = (N) => {
        N == null || N.forEach((F) => {
          var se;
          I.push(F), (se = F.children) != null && se.length && _(F.children);
        });
      };
      return _(w), I;
    }
    function j() {
      D.value.getInspectorNodeActions(s.value).then((w) => {
        l.value = w;
      });
    }
    function O() {
      D.value.getInspectorActions(s.value).then((w) => {
        i.value = w;
      });
    }
    j(), O();
    function A(w) {
      D.value.callInspectorNodeAction(s.value, w, a.value);
    }
    function k(w) {
      D.value.callInspectorAction(s.value, w);
    }
    function R(w) {
      var I;
      for (const _ in w)
        (I = w[_]) != null && I.length || delete w[_];
      return w;
    }
    function H(w) {
      D.value.getInspectorState({ inspectorId: s.value, nodeId: w }).then((I) => {
        const _ = Y(I);
        _ && (f.value = R(_), o.value = Array.from({ length: Object.keys(f.value).length }, (N, F) => `${F}`));
      });
    }
    function K() {
      f.value = {};
    }
    X(a, () => {
      K(), H(a.value);
    });
    const Z = () => {
      D.value.getInspectorTree({ inspectorId: s.value, filter: "" }).then((w) => {
        const I = Y(w);
        u.value = I, !a.value && I.length && (a.value = I[0].id, e.value = h(c.value, 1), H(I[0].id));
      });
    };
    Nt(s).toBeTruthy().then(Z);
    function E(w) {
      var _, N;
      const I = Y(w);
      !I.rootNodes.length || I.inspectorId !== s.value || (u.value = I.rootNodes, v.value.includes(a.value) || (a.value = (N = (_ = u.value) == null ? void 0 : _[0]) == null ? void 0 : N.id, e.value = h(c.value, 1)));
    }
    function P(w) {
      const I = Y(w);
      if (I.inspectorId !== s.value || !I.state || I.nodeId !== a.value)
        return;
      const { inspectorId: _, ...N } = I.state;
      f.value = R(N);
    }
    return Re(() => {
      D.functions.on(W.INSPECTOR_TREE_UPDATED, E), D.functions.on(W.INSPECTOR_STATE_UPDATED, P);
    }), ye(() => {
      D.functions.off(W.INSPECTOR_TREE_UPDATED, E), D.functions.off(W.INSPECTOR_STATE_UPDATED, P);
    }), (w, I) => (p(), m("div", S1, [
      x(Se, {
        "doc-link": d(t).homepage
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }, 8, ["doc-link"]),
      u.value.length ? (p(), z(d(Ae), {
        key: 0,
        class: "flex-1 overflow-auto"
      }, {
        default: M(() => [
          x(d(ue), {
            border: "r base",
            size: "40",
            "h-full": ""
          }, {
            default: M(() => {
              var _;
              return [
                r("div", T1, [
                  (_ = i.value) != null && _.length ? (p(), m("div", E1, [
                    r("div", V1, [
                      (p(!0), m(te, null, ae(i.value, (N, F) => J((p(), m("div", {
                        key: F,
                        class: "flex items-center gap1",
                        onClick: (se) => k(F)
                      }, [
                        x(d(yt), {
                          name: `baseline-${N.icon.replace(/\_/g, "-")}`,
                          "cursor-pointer": "",
                          op70: "",
                          "text-base": "",
                          "hover:op100": ""
                        }, null, 8, ["name"])
                      ], 8, N1)), [
                        [
                          d(Q),
                          { content: N.tooltip },
                          void 0,
                          { "bottom-end": !0 }
                        ]
                      ])), 128))
                    ])
                  ])) : U("", !0),
                  r("div", R1, [
                    x(Pe, {
                      modelValue: a.value,
                      "onUpdate:modelValue": I[0] || (I[0] = (N) => a.value = N),
                      data: u.value
                    }, null, 8, ["modelValue", "data"])
                  ])
                ])
              ];
            }),
            _: 1
          }),
          x(d(ue), { size: "60" }, {
            default: M(() => {
              var _;
              return [
                r("div", A1, [
                  (_ = l.value) != null && _.length ? (p(), m("div", D1, [
                    r("div", P1, [
                      (p(!0), m(te, null, ae(l.value, (N, F) => J((p(), m("div", {
                        key: F,
                        class: "flex items-center gap1",
                        onClick: (se) => A(F)
                      }, [
                        x(d(yt), {
                          name: `baseline-${N.icon.replace(/\_/g, "-")}`,
                          "cursor-pointer": "",
                          op70: "",
                          "text-base": "",
                          "hover:op100": ""
                        }, null, 8, ["name"])
                      ], 8, M1)), [
                        [
                          d(Q),
                          { content: N.tooltip },
                          void 0,
                          { "bottom-end": !0 }
                        ]
                      ])), 128))
                    ])
                  ])) : U("", !0),
                  a.value && !g.value ? (p(), z(De, {
                    key: 1,
                    data: f.value,
                    "node-id": a.value,
                    "inspector-id": s.value,
                    "expanded-state-id": "custom-inspector-state",
                    class: "no-scrollbar flex-1 select-none overflow-scroll"
                  }, null, 8, ["data", "node-id", "inspector-id"])) : (p(), z(Ne, { key: 2 }, {
                    default: M(() => I[1] || (I[1] = [
                      ne(" No Data ")
                    ])),
                    _: 1
                  }))
                ])
              ];
            }),
            _: 1
          })
        ]),
        _: 1
      })) : (p(), z(Ne, { key: 1 }, {
        default: M(() => I[2] || (I[2] = [
          ne(" No Data ")
        ])),
        _: 1
      }))
    ]));
  }
}), L1 = { class: "p3" }, O1 = ["onClick"], j1 = {
  key: 0,
  class: "absolute left-5px top-4.5 h10 w0 border-l-2",
  border: "solid gray2"
}, U1 = { class: "h-full flex items-center truncate pl5" }, z1 = {
  absolute: "",
  "top-5": "",
  pr2: "",
  "text-3": "",
  op40: ""
}, B1 = {
  pl2: "",
  op30: ""
}, Z1 = /* @__PURE__ */ B({
  __name: "EventList",
  props: /* @__PURE__ */ Be({
    data: {}
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(n) {
    const e = n, o = et(n, "modelValue"), t = ["#3e5770", "#42b983", "#0098c4"], s = T(null), l = V(() => {
      let i = -1, u = 0;
      return e.data.forEach((c, C) => {
        (c.groupId !== u || i === -1) && (i = (i + 1) % t.length), u = c.groupId ?? u, c.id = C, c.color = t[i];
      }), e.data;
    });
    return X(() => l.value.length, (i) => {
      var u;
      (u = s.value) == null || u.scrollToItem(i - 1);
    }, { flush: "post" }), (i, u) => (p(), m("div", L1, [
      x(d(xn), {
        ref_key: "scrollerRef",
        ref: s,
        items: l.value,
        "min-item-size": 52,
        "key-field": "id",
        "page-mode": "",
        "item-tag": "li",
        "list-tag": "ul",
        buffer: 20
      }, {
        default: M(({ item: c }) => [
          r("div", {
            class: "relative mb7 h6 cursor-pointer",
            style: Ie({ color: o.value === c.id ? c.color : "" }),
            onClick: (C) => o.value = c.id
          }, [
            r("span", {
              class: "absolute top-1.5 inline-block h3 w3 b rounded-50%",
              style: Ie({ border: `3px solid ${c.color}` })
            }, null, 4),
            c.id < i.data.length - 1 ? (p(), m("span", j1)) : U("", !0),
            r("p", U1, [
              r("span", z1, "[" + le(d(zn)(c.time)) + "]", 1),
              ne(" " + le(c.title) + " ", 1),
              r("span", B1, le(c.subtitle), 1)
            ])
          ], 12, O1)
        ]),
        _: 1
      }, 8, ["items"])
    ]));
  }
}), F1 = { class: "h-full flex flex-col" }, G1 = {
  key: 0,
  class: "flex-1 overflow-hidden"
}, K1 = {
  "h-full": "",
  "select-none": "",
  "overflow-scroll": "",
  class: "no-scrollbar"
}, W1 = {
  "h-full": "",
  "select-none": "",
  "overflow-scroll": "",
  class: "no-scrollbar"
}, dt = /* @__PURE__ */ B({
  __name: "index",
  props: {
    layerIds: {},
    docLink: {},
    githubRepoLink: {}
  },
  setup(n) {
    const e = n, { expanded: o } = me("timeline-state");
    o.value = ["0", "1"];
    const t = T([]), s = T(/* @__PURE__ */ new Map()), l = T(0), i = V(() => t.value[l.value] ?? null), u = V(() => {
      var g;
      const f = [];
      for (const b in (g = i.value) == null ? void 0 : g.data)
        f.push({
          key: b,
          type: b,
          editable: !1,
          value: i.value.data[b]
        });
      return f;
    }), c = V(() => {
      var b, h, S;
      const f = (b = i.value) == null ? void 0 : b.groupId, g = s.value.get(f);
      if (g) {
        const j = ((h = g[g.length - 1]) == null ? void 0 : h.time) - (((S = g[0]) == null ? void 0 : S.time) ?? 0);
        return [{
          key: "events",
          type: "events",
          editable: !1,
          value: g.length
        }, j && {
          key: "duration",
          type: "duration",
          editable: !1,
          value: `${j}ms`
        }].filter(Boolean);
      }
    }), C = V(() => ({ "Event Info": u.value, ...c.value && { "Group Info": c.value } }));
    function v(f) {
      var b;
      const g = f.groupId;
      g !== void 0 && (s.value.set(g, s.value.get(g) ?? []), (b = s.value.get(g)) == null || b.push(f));
    }
    function a(f) {
      const g = Y(f);
      if (!g)
        return;
      const { layerId: b, event: h } = g;
      e.layerIds.includes(b) && (t.value.push(h), v(h));
    }
    return D.functions.on(W.TIMELINE_EVENT_UPDATED, a), ye(() => {
      D.functions.off(W.TIMELINE_EVENT_UPDATED, a);
    }), (f, g) => (p(), m("div", F1, [
      x(Se, {
        "doc-link": f.docLink,
        "github-repo-link": f.githubRepoLink
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }, 8, ["doc-link", "github-repo-link"]),
      t.value.length ? (p(), m("div", G1, [
        x(d(Ae), { class: "h-full" }, {
          default: M(() => [
            x(d(ue), {
              border: "r base",
              size: "40",
              "h-full": ""
            }, {
              default: M(() => [
                r("div", K1, [
                  x(Z1, {
                    modelValue: l.value,
                    "onUpdate:modelValue": g[0] || (g[0] = (b) => l.value = b),
                    data: t.value
                  }, null, 8, ["modelValue", "data"])
                ])
              ]),
              _: 1
            }),
            x(d(ue), { size: "60" }, {
              default: M(() => [
                r("div", W1, [
                  x(De, {
                    class: "p3",
                    data: C.value,
                    "node-id": "",
                    "inspector-id": "",
                    "disable-edit": !0,
                    "expanded-state-id": "timeline-state"
                  }, null, 8, ["data"])
                ])
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ])) : (p(), z(Ne, {
        key: 1,
        class: "flex-1"
      }, {
        default: M(() => g[1] || (g[1] = [
          ne(" No events ")
        ])),
        _: 1
      }))
    ]));
  }
}), q1 = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const e = $e();
    return (o, t) => (p(), z(dt, {
      "layer-ids": d(e).timelineLayerIds,
      "doc-link": d(e).homepage
    }, null, 8, ["layer-ids", "doc-link"]));
  }
}), J1 = {
  "h-full": "",
  "w-full": ""
}, Y1 = { key: 0 }, Jl = /* @__PURE__ */ B({
  __name: "index",
  props: {
    id: {},
    pluginId: {}
  },
  emits: ["loadError"],
  setup(n, { emit: e }) {
    const o = n, t = e, s = zt(), l = T(!1), i = T(null);
    ge("pluginSettings", i);
    const u = V(() => {
      var a;
      return [
        {
          path: "/state",
          name: "State",
          component: H1,
          icon: "i-carbon-tree-view-alt"
        },
        ((a = s.value.timelineLayerIds) == null ? void 0 : a.length) && {
          path: "/timeline",
          name: "Timeline",
          component: q1,
          icon: "i-mdi:timeline-clock-outline"
        },
        {
          path: "/about",
          name: "About",
          component: d1
        },
        i.value && {
          path: "/settings",
          name: "Settings",
          component: w1,
          icon: "i-mdi:cog-outline"
        }
      ].filter(Boolean);
    }), { VirtualRouterView: c, restoreRouter: C } = ct(u, {
      defaultRoutePath: "/state"
    });
    function v() {
      l.value = !0, Re(() => {
        D.value.getInspectorInfo(o.id).then((a) => {
          if (!a) {
            t("loadError");
            return;
          }
          const f = {
            homepage: a == null ? void 0 : a.homepage,
            id: a == null ? void 0 : a.id,
            label: a == null ? void 0 : a.label,
            logo: a == null ? void 0 : a.logo,
            timelineLayerIds: a == null ? void 0 : a.timelineLayers.map((g) => g.id),
            pluginId: o.pluginId
          };
          s.value = f, C(), l.value = !1;
        }), D.value.getPluginSettings(o.pluginId).then((a) => {
          a.options ? i.value = a : i.value = null;
        });
      });
    }
    return X(() => o.id, () => {
      v();
    }, {
      immediate: !0
    }), ye(() => {
      D.value.unhighlight();
    }), (a, f) => (p(), m("div", J1, [
      l.value ? (p(), m("div", Y1, [
        x(Go)
      ])) : (p(), z(d(c), { key: 1 }))
    ]));
  }
}), X1 = {}, Q1 = {
  viewBox: "0 0 566 154",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};
function el(n, e) {
  return p(), m("svg", Q1, e[0] || (e[0] = [
    nn('<svg height="130" viewBox="0 0 319 477" width="180" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><linearGradient id="a"><stop offset="0" stop-color="#52ce63"></stop><stop offset="1" stop-color="#51a256"></stop></linearGradient><linearGradient id="b" x1="55.342075%" x2="42.816933%" xlink:href="#a" y1="0%" y2="42.862855%"></linearGradient><linearGradient id="c" x1="55.348642%" x2="42.808103%" xlink:href="#a" y1="0%" y2="42.862855%"></linearGradient><linearGradient id="d" x1="50%" x2="50%" y1="0%" y2="58.811243%"><stop offset="0" stop-color="#8ae99c"></stop><stop offset="1" stop-color="#52ce63"></stop></linearGradient><linearGradient id="e" x1="51.37763%" x2="44.584719%" y1="17.472551%" y2="100%"><stop offset="0" stop-color="#ffe56c"></stop><stop offset="1" stop-color="#ffc63a"></stop></linearGradient><g fill="none" fill-rule="evenodd" transform="translate(-34 -24)"><g transform="matrix(.99254615 .12186934 -.12186934 .99254615 33.922073 .976691)"><path d="m103.950535 258.274149c44.361599-4.360825 60.014503-40.391282 65.353094-94.699444s-30.93219-103.451001-46.020347-101.9678079c-15.088156 1.4831932-63.0385313 58.9051239-68.3771222 113.2132869-5.3385908 54.308162 4.6827754 87.814791 49.0443752 83.453965z" fill="url(#b)" transform="matrix(.70710678 -.70710678 .70710678 .70710678 -80.496332 125.892944)"></path><path d="m275.876752 258.273992c44.3616 4.360826 53.167133-29.265322 47.828542-83.573485-5.338591-54.308162-52.073133-111.6105744-67.16129-113.0937675-15.088156-1.4831931-52.57477 47.5401275-47.236179 101.8482895s22.207328 90.458137 66.568927 94.818963z" fill="url(#c)" transform="matrix(.70710678 .70710678 -.70710678 .70710678 191.403399 -141.861963)"></path><path d="m188.370027 216.876305c39.941834 0 50.95265-38.251987 50.95265-97.89874 0-59.6467532-37.367733-118.10125956-50.95265-118.10125956s-52.04735 58.45450636-52.04735 118.10125956c0 59.646753 12.105516 97.89874 52.04735 97.89874z" fill="url(#d)"></path></g><path d="m184.473473 501c83.118854 0 150.526527-24.145148 150.526527-133.645148s-67.407673-199.354852-150.526527-199.354852c-83.118855 0-150.473473 89.854852-150.473473 199.354852s67.354618 133.645148 150.473473 133.645148z" fill="url(#e)"></path><ellipse cx="260.5" cy="335" fill="#eaadcc" rx="21.5" ry="10"></ellipse><ellipse cx="102.5" cy="329" fill="#eaadcc" rx="21.5" ry="10" transform="matrix(.99254615 .12186934 -.12186934 .99254615 40.859033 -10.039292)"></ellipse><g transform="matrix(-.99939083 .0348995 .0348995 .99939083 269.284825 271.027667)"><path d="m73.1046985 58.2728794c6.7372416 4.9130333 14.3132632 6.6640587 22.7280649 5.2530761 8.4148016-1.4109825 14.5054466-5.2535769 18.2719346-11.527783" stroke="#000" stroke-linecap="round" stroke-linejoin="round" stroke-width="6" transform="matrix(.9998477 .01745241 -.01745241 .9998477 1.026464 -1.624794)"></path><path d="m154.501124 3c-5.937545 0-11.312782 2.40629805-15.203644 6.29680621-3.89094 3.89058529-6.29748 9.26545449-6.29748 15.20263179 0 5.9376888 2.406488 11.3127422 6.297291 15.2034272 3.890886 3.8907673 9.266197 6.2971348 15.203833 6.2971348 5.937109 0 11.311896-2.4063889 15.202387-6.2972348 3.890299-3.8906535 6.296489-9.2656636 6.296489-15.2033272 0-5.9371521-2.406242-11.3119781-6.296677-15.20253181-3.890469-3.89058674-9.265181-6.29690619-15.202199-6.29690619z" fill="#000"></path><path d="m154 21c0-3.865549 3.135362-7 6.999413-7 3.866399 0 7.000587 3.134451 7.000587 7s-3.134188 7-7.000587 7c-3.864051-.0011735-6.999413-3.134451-6.999413-7z" fill="#fff"></path><path d="m24.5 13c-5.9375292 0-11.312426 2.406268-15.20299427 6.2967181-3.89069464 3.8905765-6.29700573 9.2654765-6.29700573 15.2027199 0 5.9377549 2.40625962 11.3128391 6.29681766 15.2035153 3.89059104 3.8907092 9.26556184 6.2970467 15.20318234 6.2970467 5.9371249 0 11.3122514-2.406419 15.2030371-6.2973229 3.8905441-3.8906623 6.2969629-9.2656416 6.2969629-15.2032391 0-5.937086-2.4064703-11.3118811-6.297151-15.2024437-3.890763-3.8906448-9.2658154-6.2969943-15.202849-6.2969943z" fill="#000"></path><g fill="#fff"><path d="m136 24.499438c0 10.2185232 8.282911 18.500562 18.501124 18.500562 10.217089 0 18.498876-8.2820388 18.498876-18.500562 0-10.2173992-8.281787-18.499438-18.498876-18.499438-10.218213 0-18.501124 8.2820388-18.501124 18.499438zm-6 0c0-13.5311954 10.96929-24.499438 24.501124-24.499438 13.530838 0 24.498876 10.9683711 24.498876 24.499438 0 13.5319607-10.967808 24.500562-24.498876 24.500562-13.532064 0-24.501124-10.9684728-24.501124-24.500562z" fill-rule="nonzero" stroke="#fff" stroke-width="3"></path><path d="m6 34.499438c0 10.2185232 8.2817873 18.500562 18.5 18.500562 10.2170889 0 18.5-8.2820388 18.5-18.500562 0-10.2173992-8.2829111-18.499438-18.5-18.499438-10.2182127 0-18.5 8.2820388-18.5 18.499438zm-6 0c0-13.531297 10.9682681-24.499438 24.5-24.499438 13.5309398 0 24.5 10.9684728 24.5 24.499438 0 13.5318591-10.96883 24.500562-24.5 24.500562-13.531962 0-24.5-10.9683711-24.5-24.500562z" fill-rule="nonzero" stroke="#fff" stroke-width="3"></path><path d="m24 31c0-3.865549 3.134451-7 7-7s7 3.134451 7 7-3.134451 7-7 7-7-3.134451-7-7z"></path></g></g><g stroke-linecap="round" stroke-width="11"><g stroke="#ecb732"><path d="m70.5 377.5 74 77"></path><path d="m134.5 386.5-47 50"></path></g><g stroke="#ecb732" transform="matrix(-1 0 0 1 298 377)"><path d="m.5.5 74 77"></path><path d="m64.5 9.5-47 50"></path></g><g stroke="#ffc73b" transform="matrix(0 1 -1 0 215 207)"><path d="m.5.5 49 49"></path><path d="m.5 10.5 49 49" transform="matrix(-1 0 0 1 50 0)"></path></g></g></g></svg><path d="M152.571 118V46.1333H176.8C183.576 46.1333 189.77 47.468 195.383 50.1373C200.995 52.8067 205.478 56.8449 208.832 62.252C212.186 67.5907 213.863 74.1956 213.863 82.0667C213.863 89.8693 212.186 96.4742 208.832 101.881C205.478 107.288 200.995 111.327 195.383 113.996C189.77 116.665 183.576 118 176.8 118H152.571ZM175.157 104.653C181.796 104.653 187.306 102.874 191.687 99.3147C196.067 95.6871 198.257 89.9378 198.257 82.0667C198.257 74.1956 196.067 68.4462 191.687 64.8187C187.306 61.1911 181.796 59.3773 175.157 59.3773H167.971V104.653H175.157ZM270.18 100.444C269.153 105.988 266.381 110.437 261.864 113.791C257.347 117.144 251.905 118.821 245.54 118.821C239.928 118.821 235.102 117.555 231.064 115.023C227.094 112.422 224.083 109.068 222.029 104.961C219.976 100.855 218.949 96.5427 218.949 92.0253C218.949 87.4396 219.873 83.1276 221.721 79.0893C223.638 75.0511 226.513 71.7658 230.345 69.2333C234.247 66.7009 238.969 65.4347 244.513 65.4347C250.263 65.4347 255.054 66.6667 258.887 69.1307C262.72 71.5947 265.526 74.7431 267.305 78.576C269.153 82.4089 270.077 86.4471 270.077 90.6907C270.077 92.2649 270.009 93.8049 269.872 95.3107H232.707C233.254 99.0067 234.623 101.916 236.813 104.037C239.072 106.091 241.981 107.117 245.54 107.117C248.415 107.117 250.776 106.57 252.624 105.475C254.472 104.311 255.67 102.634 256.217 100.444H270.18ZM244.513 75.9067C241.228 75.9067 238.627 76.7622 236.711 78.4733C234.794 80.116 233.528 82.7169 232.912 86.276H255.807C255.601 83.2644 254.506 80.8004 252.521 78.884C250.537 76.8991 247.867 75.9067 244.513 75.9067ZM302.77 118H289.629L270.738 66.256H285.317L296.2 99.7253L306.98 66.256H321.661L302.77 118ZM359.731 118H344.331V59.3773H323.079V46.1333H380.983V59.3773H359.731V118ZM402.364 118.821C397.163 118.821 392.474 117.692 388.299 115.433C384.124 113.106 380.873 109.924 378.546 105.885C376.219 101.779 375.055 97.1929 375.055 92.128C375.055 87.0631 376.219 82.5116 378.546 78.4733C380.873 74.3667 384.124 71.184 388.299 68.9253C392.474 66.5982 397.163 65.4347 402.364 65.4347C407.566 65.4347 412.255 66.5982 416.43 68.9253C420.605 71.184 423.856 74.3667 426.183 78.4733C428.51 82.5116 429.674 87.0631 429.674 92.128C429.674 97.1929 428.51 101.779 426.183 105.885C423.856 109.924 420.605 113.106 416.43 115.433C412.255 117.692 407.566 118.821 402.364 118.821ZM402.364 106.501C406.197 106.501 409.311 105.167 411.707 102.497C414.103 99.828 415.3 96.3716 415.3 92.128C415.3 87.816 414.103 84.3253 411.707 81.656C409.311 78.9867 406.197 77.652 402.364 77.652C398.531 77.652 395.417 78.9867 393.022 81.656C390.626 84.3253 389.428 87.816 389.428 92.128C389.428 96.3716 390.626 99.828 393.022 102.497C395.417 105.167 398.531 106.501 402.364 106.501ZM462.12 118.821C456.918 118.821 452.229 117.692 448.054 115.433C443.879 113.106 440.628 109.924 438.301 105.885C435.974 101.779 434.81 97.1929 434.81 92.128C434.81 87.0631 435.974 82.5116 438.301 78.4733C440.628 74.3667 443.879 71.184 448.054 68.9253C452.229 66.5982 456.918 65.4347 462.12 65.4347C467.321 65.4347 472.01 66.5982 476.185 68.9253C480.36 71.184 483.611 74.3667 485.938 78.4733C488.265 82.5116 489.429 87.0631 489.429 92.128C489.429 97.1929 488.265 101.779 485.938 105.885C483.611 109.924 480.36 113.106 476.185 115.433C472.01 117.692 467.321 118.821 462.12 118.821ZM462.12 106.501C465.952 106.501 469.067 105.167 471.462 102.497C473.858 99.828 475.056 96.3716 475.056 92.128C475.056 87.816 473.858 84.3253 471.462 81.656C469.067 78.9867 465.952 77.652 462.12 77.652C458.287 77.652 455.172 78.9867 452.777 81.656C450.381 84.3253 449.184 87.816 449.184 92.128C449.184 96.3716 450.381 99.828 452.777 102.497C455.172 105.167 458.287 106.501 462.12 106.501ZM512.327 118H498.056V43.772H512.327V118ZM542.494 118.821C536.129 118.821 530.961 117.179 526.991 113.893C523.09 110.54 521.002 106.091 520.729 100.547H533.049C533.322 102.874 534.281 104.722 535.923 106.091C537.634 107.391 539.825 108.041 542.494 108.041C544.684 108.041 546.464 107.562 547.833 106.604C549.27 105.646 549.989 104.448 549.989 103.011C549.989 101.094 549.167 99.7596 547.525 99.0067C545.882 98.2538 543.281 97.5693 539.722 96.9533C536.026 96.2689 533.014 95.516 530.687 94.6947C528.36 93.8733 526.341 92.4018 524.63 90.28C522.987 88.0898 522.166 85.0098 522.166 81.04C522.166 78.0284 522.953 75.3591 524.527 73.032C526.17 70.6364 528.394 68.7884 531.201 67.488C534.007 66.1191 537.155 65.4347 540.646 65.4347C546.874 65.4347 551.905 66.9747 555.738 70.0547C559.639 73.1347 561.727 77.2071 562.001 82.272H549.578C549.304 80.1502 548.312 78.5076 546.601 77.344C544.958 76.112 543.11 75.496 541.057 75.496C539.003 75.496 537.361 75.9409 536.129 76.8307C534.897 77.7204 534.281 78.9524 534.281 80.5267C534.281 82.4431 535.068 83.7436 536.642 84.428C538.285 85.044 540.851 85.5916 544.342 86.0707C548.106 86.6182 551.186 87.3027 553.582 88.124C556.046 88.8769 558.168 90.3827 559.947 92.6413C561.727 94.9 562.617 98.1853 562.617 102.497C562.617 107.425 560.769 111.395 557.073 114.407C553.445 117.35 548.585 118.821 542.494 118.821Z" fill="currentColor"></path>', 2)
  ]));
}
const tl = /* @__PURE__ */ be(X1, [["render", el]]), nl = {
  "h-full": "",
  "w-full": "",
  flex: "",
  "items-center": ""
}, ol = {
  flex: "~ col gap2",
  ma: "",
  "px-5": ""
}, ll = {
  flex: "~ col",
  "mt-20": "",
  "items-center": ""
}, sl = {
  flex: "~",
  "mt--10": "",
  "items-center": "",
  "justify-center": ""
}, al = { flex: "~ gap2 wrap" }, il = /* @__PURE__ */ B({
  __name: "About",
  setup(n) {
    const e = Me();
    return (o, t) => (p(), m("div", nl, [
      r("div", ol, [
        t[5] || (t[5] = r("div", { "flex-auto": "" }, null, -1)),
        r("div", ll, [
          r("div", sl, [
            x(tl, { "h-18": "" })
          ]),
          t[2] || (t[2] = r("div", {
            mb6: "",
            "mt--1": "",
            "text-center": "",
            "text-sm": "",
            flex: "~ gap-1"
          }, [
            r("span", { op40: "" }, " Pinia DevTools ")
          ], -1))
        ]),
        t[6] || (t[6] = r("div", { "flex-auto": "" }, null, -1)),
        r("div", al, [
          r("div", {
            flex: "~ col auto",
            "min-w-40": "",
            p4: "",
            "theme-card-lime": "",
            onClick: t[0] || (t[0] = (s) => d(e).push("/store"))
          }, t[3] || (t[3] = [
            r("div", {
              "i-carbon-tree-view-alt": "",
              "text-3xl": ""
            }, null, -1),
            r("code", null, "Store", -1)
          ])),
          r("div", {
            flex: "~ col auto",
            "min-w-40": "",
            p4: "",
            "theme-card-lime": "",
            onClick: t[1] || (t[1] = (s) => d(e).push("/timeline"))
          }, t[4] || (t[4] = [
            r("div", {
              "i-mdi:timeline-clock-outline": "",
              "text-3xl": ""
            }, null, -1),
            r("div", null, "Timeline", -1)
          ]))
        ]),
        t[7] || (t[7] = r("div", {
          flex: "~ gap-6 wrap",
          "mt-5": "",
          "items-center": "",
          "justify-center": ""
        }, [
          r("a", {
            href: "https://github.com/vuejs/pinia",
            target: "_blank",
            flex: "~ gap1",
            "items-center": "",
            op50: "",
            hover: "op100 text-blue",
            transition: ""
          }, [
            r("div", { "i-carbon-star": "" }),
            ne(" Star on GitHub ")
          ]),
          r("a", {
            href: "https://pinia.vuejs.org/",
            target: "_blank",
            flex: "~ gap1",
            "items-center": "",
            op50: "",
            hover: "op100 text-yellow",
            transition: ""
          }, [
            r("div", { "i-carbon-document": "" }),
            ne(" View Documentation ")
          ])
        ], -1)),
        t[8] || (t[8] = r("div", { "flex-auto": "" }, null, -1))
      ])
    ]));
  }
}), rl = { class: "h-full flex flex-col" }, ul = /* @__PURE__ */ B({
  __name: "Settings",
  setup(n) {
    const e = ve("pluginSettings"), o = V(() => e.value.options), t = V(() => e.value.values);
    function s(l) {
      e.value = l;
    }
    return (l, i) => (p(), m("div", rl, [
      x(Se, {
        "doc-link": "https://pinia.vuejs.org/",
        "github-repo-link": "https://github.com/vuejs/pinia"
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }),
      x(Zt, {
        "plugin-id": "dev.esm.pinia",
        options: o.value,
        values: t.value,
        onUpdate: s
      }, null, 8, ["options", "values"])
    ]));
  }
}), cl = { class: "h-full flex flex-col" }, dl = { class: "h-full flex flex-col p2" }, pl = {
  key: 0,
  class: "mb-1 flex justify-end pb-1",
  border: "b dashed base"
}, fl = { class: "flex items-center gap-2 px-1" }, vl = ["onClick"], ml = { class: "no-scrollbar flex-1 select-none overflow-scroll" }, hl = { class: "h-full flex flex-col p2" }, gl = {
  key: 0,
  class: "mb-1 flex justify-end pb-1",
  border: "b dashed base"
}, Cl = { class: "flex items-center gap-2 px-1" }, _l = ["onClick"], pe = "pinia", xl = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const { expanded: e } = me(), { expanded: o } = me("pinia-store-state"), t = T([]), s = T([]), l = T(""), i = T([]), u = V(() => {
      var E, P;
      return (E = i.value) != null && E.length ? f((P = i.value) == null ? void 0 : P[0]) : [];
    }), c = V(() => b(i.value)), C = V(() => c.value.map((E) => E.id)), v = T({}), a = V(() => {
      var E, P;
      return !((E = v.value.state) != null && E.length) && !((P = v.value.getters) != null && P.length);
    });
    function f(E, P = [], w = []) {
      var I;
      return P.push(E.id), ((I = E.children) == null ? void 0 : I.length) === 0 && w.push([...P]), Array.isArray(E.children) && E.children.forEach((_) => {
        f(_, P, w);
      }), P.pop(), w;
    }
    function g(E, P) {
      const w = [];
      return E == null || E.forEach((I) => {
        w.push(...I.slice(0, P + 1));
      }), [...new Set(w)];
    }
    function b(E) {
      const P = [], w = (I) => {
        I == null || I.forEach((_) => {
          var N;
          P.push(_), (N = _.children) != null && N.length && w(_.children);
        });
      };
      return w(E), P;
    }
    function h() {
      D.value.getInspectorNodeActions(pe).then((E) => {
        t.value = E;
      });
    }
    function S() {
      D.value.getInspectorActions(pe).then((E) => {
        s.value = E;
      });
    }
    h(), S();
    function j(E) {
      D.value.callInspectorNodeAction(pe, E, l.value);
    }
    function O(E) {
      D.value.callInspectorAction(pe, E);
    }
    function A(E) {
      var P;
      for (const w in E)
        (P = E[w]) != null && P.length || delete E[w];
      return E;
    }
    function k(E) {
      D.value.getInspectorState({ inspectorId: pe, nodeId: E }).then((P) => {
        const w = Y(P);
        w && (v.value = A(w), o.value = Array.from({ length: Object.keys(v.value).length }, (I, _) => `${_}`));
      });
    }
    function R() {
      v.value = {};
    }
    X(l, () => {
      R(), k(l.value);
    }), (() => {
      D.value.getInspectorTree({ inspectorId: pe, filter: "" }).then((E) => {
        const P = Y(E);
        i.value = P, !l.value && P.length && (l.value = P[0].id, k(P[0].id), e.value = g(u.value, 1));
      });
    })();
    function K(E) {
      const P = Y(E);
      P.inspectorId !== pe || !P.rootNodes.length || (i.value = P.rootNodes, C.value.includes(l.value) || (l.value = P.rootNodes[0].id, e.value = g(u.value, 1), k(P.rootNodes[0].id)));
    }
    function Z(E) {
      const P = Y(E);
      if (P.inspectorId !== pe)
        return;
      const w = P.state;
      v.value = A({
        state: w.state,
        getters: w.getters
      });
    }
    return D.functions.on(W.INSPECTOR_TREE_UPDATED, K), D.functions.on(W.INSPECTOR_STATE_UPDATED, Z), ye(() => {
      D.functions.off(W.INSPECTOR_TREE_UPDATED, K), D.functions.off(W.INSPECTOR_STATE_UPDATED, Z);
    }), (E, P) => (p(), m("div", cl, [
      x(Se, {
        "doc-link": "https://pinia.vuejs.org/",
        "github-repo-link": "https://github.com/vuejs/pinia"
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }),
      x(d(Ae), { class: "flex-1 overflow-auto" }, {
        default: M(() => [
          x(d(ue), {
            border: "r base",
            size: "40",
            "h-full": ""
          }, {
            default: M(() => {
              var w;
              return [
                r("div", dl, [
                  (w = s.value) != null && w.length ? (p(), m("div", pl, [
                    r("div", fl, [
                      (p(!0), m(te, null, ae(s.value, (I, _) => J((p(), m("div", {
                        key: _,
                        class: "flex items-center gap1",
                        onClick: (N) => O(_)
                      }, [
                        r("i", {
                          class: ie(`i-ic-baseline-${I.icon.replace(/\_/g, "-")}`),
                          "cursor-pointer": "",
                          op70: "",
                          "text-base": "",
                          "hover:op100": ""
                        }, null, 2)
                      ], 8, vl)), [
                        [
                          d(Q),
                          { content: I.tooltip },
                          void 0,
                          { "bottom-end": !0 }
                        ]
                      ])), 128))
                    ])
                  ])) : U("", !0),
                  r("div", ml, [
                    x(Pe, {
                      modelValue: l.value,
                      "onUpdate:modelValue": P[0] || (P[0] = (I) => l.value = I),
                      data: i.value
                    }, null, 8, ["modelValue", "data"])
                  ])
                ])
              ];
            }),
            _: 1
          }),
          x(d(ue), { size: "60" }, {
            default: M(() => {
              var w;
              return [
                r("div", hl, [
                  (w = t.value) != null && w.length ? (p(), m("div", gl, [
                    r("div", Cl, [
                      (p(!0), m(te, null, ae(t.value, (I, _) => J((p(), m("div", {
                        key: _,
                        class: "flex items-center gap1",
                        onClick: (N) => j(_)
                      }, [
                        r("i", {
                          class: ie(`i-ic-baseline-${I.icon.replace(/\_/g, "-")}`),
                          "cursor-pointer": "",
                          op70: "",
                          "text-base": "",
                          "hover:op100": ""
                        }, null, 2)
                      ], 8, _l)), [
                        [
                          d(Q),
                          { content: I.tooltip },
                          void 0,
                          { "bottom-end": !0 }
                        ]
                      ])), 128))
                    ])
                  ])) : U("", !0),
                  l.value && !a.value ? (p(), z(De, {
                    key: 1,
                    class: "no-scrollbar flex-1 select-none overflow-scroll",
                    data: v.value,
                    "node-id": l.value,
                    "inspector-id": pe,
                    "expanded-state-id": "pinia-store-state"
                  }, null, 8, ["data", "node-id"])) : (p(), z(Ne, { key: 2 }, {
                    default: M(() => P[1] || (P[1] = [
                      ne(" No Data ")
                    ])),
                    _: 1
                  }))
                ])
              ];
            }),
            _: 1
          })
        ]),
        _: 1
      })
    ]));
  }
}), yl = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const e = ["pinia:mutations"];
    return (o, t) => (p(), z(dt, {
      "layer-ids": e,
      "doc-link": "https://pinia.vuejs.org/",
      "github-repo-link": "https://github.com/vuejs/pinia"
    }));
  }
}), bl = {
  "h-full": "",
  "w-full": ""
}, Yl = /* @__PURE__ */ B({
  __name: "index",
  setup(n) {
    const e = T(null);
    ge("pluginSettings", e);
    const o = V(() => [
      {
        path: "/store",
        name: "Store",
        component: xl,
        icon: "i-carbon-tree-view-alt"
      },
      {
        path: "/timeline",
        name: "Timeline",
        component: yl,
        icon: "i-mdi:timeline-clock-outline"
      },
      {
        path: "/",
        name: "About",
        component: il,
        icon: "i-logos-pinia"
      },
      e.value && {
        path: "/settings",
        name: "Settings",
        component: ul,
        icon: "i-mdi:cog-outline"
      }
    ].filter(Boolean)), { VirtualRouterView: t, restoreRouter: s } = ct(o, {
      defaultRoutePath: "/store"
    });
    return Re(() => {
      D.value.getPluginSettings("dev.esm.pinia").then((i) => {
        i.options ? e.value = i : e.value = null;
      });
    }), (l, i) => (p(), m("div", bl, [
      x(d(t))
    ]));
  }
}), wl = {}, kl = {
  viewBox: "0 0 566 154",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};
function Il(n, e) {
  return p(), m("svg", kl, e[0] || (e[0] = [
    r("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      width: "8em",
      height: "8em",
      viewBox: "0 -4 24 24"
    }, [
      r("path", {
        fill: "#3dd68c",
        d: "M4 15V8.5a4.5 4.5 0 0 1 9 0v7a2.5 2.5 0 0 0 5 0V8.83a3.001 3.001 0 1 1 2 0v6.67a4.5 4.5 0 1 1-9 0v-7a2.5 2.5 0 0 0-5 0V15h3l-4 5l-4-5zm15-8a1 1 0 1 0 0-2a1 1 0 0 0 0 2"
      })
    ], -1),
    r("path", {
      d: "M152.571 118V46.1333H176.8C183.576 46.1333 189.77 47.468 195.383 50.1373C200.995 52.8067 205.478 56.8449 208.832 62.252C212.186 67.5907 213.863 74.1956 213.863 82.0667C213.863 89.8693 212.186 96.4742 208.832 101.881C205.478 107.288 200.995 111.327 195.383 113.996C189.77 116.665 183.576 118 176.8 118H152.571ZM175.157 104.653C181.796 104.653 187.306 102.874 191.687 99.3147C196.067 95.6871 198.257 89.9378 198.257 82.0667C198.257 74.1956 196.067 68.4462 191.687 64.8187C187.306 61.1911 181.796 59.3773 175.157 59.3773H167.971V104.653H175.157ZM270.18 100.444C269.153 105.988 266.381 110.437 261.864 113.791C257.347 117.144 251.905 118.821 245.54 118.821C239.928 118.821 235.102 117.555 231.064 115.023C227.094 112.422 224.083 109.068 222.029 104.961C219.976 100.855 218.949 96.5427 218.949 92.0253C218.949 87.4396 219.873 83.1276 221.721 79.0893C223.638 75.0511 226.513 71.7658 230.345 69.2333C234.247 66.7009 238.969 65.4347 244.513 65.4347C250.263 65.4347 255.054 66.6667 258.887 69.1307C262.72 71.5947 265.526 74.7431 267.305 78.576C269.153 82.4089 270.077 86.4471 270.077 90.6907C270.077 92.2649 270.009 93.8049 269.872 95.3107H232.707C233.254 99.0067 234.623 101.916 236.813 104.037C239.072 106.091 241.981 107.117 245.54 107.117C248.415 107.117 250.776 106.57 252.624 105.475C254.472 104.311 255.67 102.634 256.217 100.444H270.18ZM244.513 75.9067C241.228 75.9067 238.627 76.7622 236.711 78.4733C234.794 80.116 233.528 82.7169 232.912 86.276H255.807C255.601 83.2644 254.506 80.8004 252.521 78.884C250.537 76.8991 247.867 75.9067 244.513 75.9067ZM302.77 118H289.629L270.738 66.256H285.317L296.2 99.7253L306.98 66.256H321.661L302.77 118ZM359.731 118H344.331V59.3773H323.079V46.1333H380.983V59.3773H359.731V118ZM402.364 118.821C397.163 118.821 392.474 117.692 388.299 115.433C384.124 113.106 380.873 109.924 378.546 105.885C376.219 101.779 375.055 97.1929 375.055 92.128C375.055 87.0631 376.219 82.5116 378.546 78.4733C380.873 74.3667 384.124 71.184 388.299 68.9253C392.474 66.5982 397.163 65.4347 402.364 65.4347C407.566 65.4347 412.255 66.5982 416.43 68.9253C420.605 71.184 423.856 74.3667 426.183 78.4733C428.51 82.5116 429.674 87.0631 429.674 92.128C429.674 97.1929 428.51 101.779 426.183 105.885C423.856 109.924 420.605 113.106 416.43 115.433C412.255 117.692 407.566 118.821 402.364 118.821ZM402.364 106.501C406.197 106.501 409.311 105.167 411.707 102.497C414.103 99.828 415.3 96.3716 415.3 92.128C415.3 87.816 414.103 84.3253 411.707 81.656C409.311 78.9867 406.197 77.652 402.364 77.652C398.531 77.652 395.417 78.9867 393.022 81.656C390.626 84.3253 389.428 87.816 389.428 92.128C389.428 96.3716 390.626 99.828 393.022 102.497C395.417 105.167 398.531 106.501 402.364 106.501ZM462.12 118.821C456.918 118.821 452.229 117.692 448.054 115.433C443.879 113.106 440.628 109.924 438.301 105.885C435.974 101.779 434.81 97.1929 434.81 92.128C434.81 87.0631 435.974 82.5116 438.301 78.4733C440.628 74.3667 443.879 71.184 448.054 68.9253C452.229 66.5982 456.918 65.4347 462.12 65.4347C467.321 65.4347 472.01 66.5982 476.185 68.9253C480.36 71.184 483.611 74.3667 485.938 78.4733C488.265 82.5116 489.429 87.0631 489.429 92.128C489.429 97.1929 488.265 101.779 485.938 105.885C483.611 109.924 480.36 113.106 476.185 115.433C472.01 117.692 467.321 118.821 462.12 118.821ZM462.12 106.501C465.952 106.501 469.067 105.167 471.462 102.497C473.858 99.828 475.056 96.3716 475.056 92.128C475.056 87.816 473.858 84.3253 471.462 81.656C469.067 78.9867 465.952 77.652 462.12 77.652C458.287 77.652 455.172 78.9867 452.777 81.656C450.381 84.3253 449.184 87.816 449.184 92.128C449.184 96.3716 450.381 99.828 452.777 102.497C455.172 105.167 458.287 106.501 462.12 106.501ZM512.327 118H498.056V43.772H512.327V118ZM542.494 118.821C536.129 118.821 530.961 117.179 526.991 113.893C523.09 110.54 521.002 106.091 520.729 100.547H533.049C533.322 102.874 534.281 104.722 535.923 106.091C537.634 107.391 539.825 108.041 542.494 108.041C544.684 108.041 546.464 107.562 547.833 106.604C549.27 105.646 549.989 104.448 549.989 103.011C549.989 101.094 549.167 99.7596 547.525 99.0067C545.882 98.2538 543.281 97.5693 539.722 96.9533C536.026 96.2689 533.014 95.516 530.687 94.6947C528.36 93.8733 526.341 92.4018 524.63 90.28C522.987 88.0898 522.166 85.0098 522.166 81.04C522.166 78.0284 522.953 75.3591 524.527 73.032C526.17 70.6364 528.394 68.7884 531.201 67.488C534.007 66.1191 537.155 65.4347 540.646 65.4347C546.874 65.4347 551.905 66.9747 555.738 70.0547C559.639 73.1347 561.727 77.2071 562.001 82.272H549.578C549.304 80.1502 548.312 78.5076 546.601 77.344C544.958 76.112 543.11 75.496 541.057 75.496C539.003 75.496 537.361 75.9409 536.129 76.8307C534.897 77.7204 534.281 78.9524 534.281 80.5267C534.281 82.4431 535.068 83.7436 536.642 84.428C538.285 85.044 540.851 85.5916 544.342 86.0707C548.106 86.6182 551.186 87.3027 553.582 88.124C556.046 88.8769 558.168 90.3827 559.947 92.6413C561.727 94.9 562.617 98.1853 562.617 102.497C562.617 107.425 560.769 111.395 557.073 114.407C553.445 117.35 548.585 118.821 542.494 118.821Z",
      fill: "currentColor"
    }, null, -1)
  ]));
}
const $l = /* @__PURE__ */ be(wl, [["render", Il]]), Sl = {
  "h-full": "",
  "w-full": "",
  flex: "",
  "items-center": ""
}, Tl = {
  flex: "~ col gap2",
  ma: "",
  "px-5": ""
}, El = {
  flex: "~ col",
  "mt-20": "",
  "items-center": ""
}, Vl = {
  flex: "~",
  "mt--10": "",
  "items-center": "",
  "justify-center": ""
}, Nl = { flex: "~ gap2 wrap" }, Rl = /* @__PURE__ */ B({
  __name: "About",
  setup(n) {
    const e = Me();
    return (o, t) => (p(), m("div", Sl, [
      r("div", Tl, [
        t[5] || (t[5] = r("div", { "flex-auto": "" }, null, -1)),
        r("div", El, [
          r("div", Vl, [
            x($l, { "h-18": "" })
          ]),
          t[2] || (t[2] = r("div", {
            mb6: "",
            "mt--1": "",
            "text-center": "",
            "text-sm": "",
            flex: "~ gap-1"
          }, [
            r("span", { op40: "" }, " Router DevTools ")
          ], -1))
        ]),
        t[6] || (t[6] = r("div", { "flex-auto": "" }, null, -1)),
        r("div", Nl, [
          r("div", {
            flex: "~ col auto",
            "min-w-40": "",
            p4: "",
            "theme-card-lime": "",
            onClick: t[0] || (t[0] = (s) => d(e).push("/routes"))
          }, t[3] || (t[3] = [
            r("div", {
              "i-carbon-tree-view-alt": "",
              "text-3xl": ""
            }, null, -1),
            r("code", null, "Routes", -1)
          ])),
          r("div", {
            flex: "~ col auto",
            "min-w-40": "",
            p4: "",
            "theme-card-lime": "",
            onClick: t[1] || (t[1] = (s) => d(e).push("/timeline"))
          }, t[4] || (t[4] = [
            r("div", {
              "i-mdi:timeline-clock-outline": "",
              "text-3xl": ""
            }, null, -1),
            r("div", null, "Timeline", -1)
          ]))
        ]),
        t[7] || (t[7] = r("div", {
          flex: "~ gap-6 wrap",
          "mt-5": "",
          "items-center": "",
          "justify-center": ""
        }, [
          r("a", {
            href: "https://github.com/vuejs/router",
            target: "_blank",
            flex: "~ gap1",
            "items-center": "",
            op50: "",
            hover: "op100 text-blue",
            transition: ""
          }, [
            r("div", { "i-carbon-star": "" }),
            ne(" Star on GitHub ")
          ]),
          r("a", {
            href: "https://router.vuejs.org/",
            target: "_blank",
            flex: "~ gap1",
            "items-center": "",
            op50: "",
            hover: "op100 text-yellow",
            transition: ""
          }, [
            r("div", { "i-carbon-document": "" }),
            ne(" View Documentation ")
          ])
        ], -1)),
        t[8] || (t[8] = r("div", { "flex-auto": "" }, null, -1))
      ])
    ]));
  }
}), Al = { class: "h-full flex flex-col" }, Dl = {
  "h-full": "",
  "select-none": "",
  "overflow-scroll": "",
  p2: "",
  class: "no-scrollbar"
}, Pl = {
  "h-full": "",
  "select-none": "",
  "overflow-scroll": "",
  class: "no-scrollbar"
}, Ml = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const { expanded: e } = me(), { expanded: o } = me("routes-state"), t = $e(), s = V(() => t.value.id), l = T(""), i = T([]), u = V(() => {
      var k, R;
      return (k = i.value) != null && k.length ? a((R = i.value) == null ? void 0 : R[0]) : [];
    }), c = V(() => g(i.value)), C = V(() => c.value.map((k) => k.id)), v = T({});
    function a(k, R = [], H = []) {
      var K;
      return R.push(k.id), ((K = k.children) == null ? void 0 : K.length) === 0 && H.push([...R]), Array.isArray(k.children) && k.children.forEach((Z) => {
        a(Z, R, H);
      }), R.pop(), H;
    }
    function f(k, R) {
      const H = [];
      return k == null || k.forEach((K) => {
        H.push(...K.slice(0, R + 1));
      }), [...new Set(H)];
    }
    function g(k) {
      const R = [], H = (K) => {
        K == null || K.forEach((Z) => {
          var E;
          R.push(Z), (E = Z.children) != null && E.length && H(Z.children);
        });
      };
      return H(k), R;
    }
    function b(k) {
      var R;
      for (const H in k)
        (R = k[H]) != null && R.length || delete k[H];
      return k;
    }
    function h(k) {
      D.value.getInspectorState({ inspectorId: s.value, nodeId: k }).then((R) => {
        const H = Y(R);
        H && (v.value = b(H), o.value = Array.from({ length: Object.keys(v.value).length }, (K, Z) => `${Z}`));
      });
    }
    function S() {
      v.value = {};
    }
    X(l, () => {
      S(), h(l.value);
    });
    const j = () => {
      D.value.getInspectorTree({ inspectorId: s.value, filter: "" }).then((k) => {
        const R = Y(k);
        i.value = R, !l.value && R.length && (l.value = R[0].id, h(R[0].id), e.value = f(u.value, 1));
      });
    };
    Nt(s).toBeTruthy().then(j);
    function O(k) {
      const R = Y(k);
      R.inspectorId !== s.value || !R.rootNodes.length || (i.value = R.rootNodes, C.value.includes(l.value) || (l.value = R.rootNodes[0].id, e.value = f(u.value, 1), h(R.rootNodes[0].id)));
    }
    function A(k) {
      const R = Y(k);
      if (R.inspectorId !== s.value)
        return;
      const H = R.state;
      v.value = b(H);
    }
    return D.functions.on(W.INSPECTOR_TREE_UPDATED, O), D.functions.on(W.INSPECTOR_STATE_UPDATED, A), ye(() => {
      D.functions.off(W.INSPECTOR_TREE_UPDATED, O), D.functions.off(W.INSPECTOR_STATE_UPDATED, A);
    }), (k, R) => (p(), m("div", Al, [
      x(Se, {
        "doc-link": "https://router.vuejs.org/",
        "github-repo-link": "https://github.com/vuejs/router"
      }, {
        default: M(() => [
          x(Te)
        ]),
        _: 1
      }),
      x(d(Ae), { class: "flex-1 overflow-auto" }, {
        default: M(() => [
          x(d(ue), {
            border: "r base",
            size: "40",
            "h-full": ""
          }, {
            default: M(() => [
              r("div", Dl, [
                x(Pe, {
                  modelValue: l.value,
                  "onUpdate:modelValue": R[0] || (R[0] = (H) => l.value = H),
                  data: i.value
                }, null, 8, ["modelValue", "data"])
              ])
            ]),
            _: 1
          }),
          x(d(ue), { size: "60" }, {
            default: M(() => [
              r("div", Pl, [
                l.value ? (p(), z(De, {
                  key: 0,
                  class: "p3",
                  data: v.value,
                  "node-id": "",
                  "inspector-id": "router",
                  "expanded-state-id": "routes-state"
                }, null, 8, ["data"])) : (p(), z(Ne, { key: 1 }, {
                  default: M(() => R[1] || (R[1] = [
                    ne(" No Data ")
                  ])),
                  _: 1
                }))
              ])
            ]),
            _: 1
          })
        ]),
        _: 1
      })
    ]));
  }
}), Hl = /* @__PURE__ */ B({
  __name: "Index",
  setup(n) {
    const e = $e();
    return (o, t) => (p(), z(dt, {
      "layer-ids": d(e).timelineLayerIds,
      "doc-link": "https://router.vuejs.org/",
      "github-repo-link": "https://github.com/vuejs/router"
    }, null, 8, ["layer-ids"]));
  }
}), Ll = {
  "h-full": "",
  "w-full": ""
}, Xl = /* @__PURE__ */ B({
  __name: "index",
  props: {
    id: {}
  },
  setup(n) {
    const e = n, o = zt(), t = T(!1), { VirtualRouterView: s, restoreRouter: l } = ct([
      {
        path: "/routes",
        name: "Routes",
        component: Ml,
        icon: "i-carbon-tree-view-alt"
      },
      {
        path: "/timeline",
        name: "Timeline",
        component: Hl,
        icon: "i-mdi:timeline-clock-outline"
      },
      {
        path: "/about",
        name: "About",
        component: Rl,
        icon: "i-ri-route-line"
      }
    ], {
      defaultRoutePath: "/routes"
    });
    function i() {
      t.value = !0, Re(() => {
        D.value.getInspectorInfo(e.id).then((u) => {
          if (!u)
            return;
          const c = {
            homepage: u == null ? void 0 : u.homepage,
            id: u == null ? void 0 : u.id,
            label: u == null ? void 0 : u.label,
            logo: u == null ? void 0 : u.logo,
            timelineLayerIds: u == null ? void 0 : u.timelineLayers.map((C) => C.id)
          };
          o.value = c, l(), t.value = !1;
        });
      });
    }
    return X(() => e.id, (u) => {
      u && i();
    }), (u, c) => (p(), m("div", Ll, [
      x(d(s))
    ]));
  }
});
export {
  ql as Components,
  Jl as CustomInspector,
  Yl as Pinia,
  Xl as Router,
  Wl as useCustomInspector
};
